---
title: "GeneralityRpf"
author: "Venus Kuo"
date: "June 30, 2017"
output: html_document
---
# Based on growthcurve_code.R Written by: M. Larsen (2013/07/18) and M. Muscarella (2015/11/24) # 


## 1) Questions

1. How do bacteria other than KBS0714 respond to Rpf?

2. Is there a phylogenetic signal to Rpf response? 


## 2) Methods

Microbial Growth Curves on KBS enviromental strains

1.  701(-) Bacteriodetes, Sphingobacteriaceae, Pedobacter Gram -
2.  702 (+),Actinobacteria, Micrococcaceae, Arthrobacter Gram +
3.  703 (+),Actinobacteria, Micrococcaceae, Arthrobacter Gram +
4.  705 (-), Proteobacteria, Rhodospirillaceae, Azospirillium Gram - KmR. Tc and Gm inhibited growth
5.  706 (+),Actinobact., Mycobacteriaceae, Mycobacterium Gram +
6.  710 (-), Proteobacteria, Pseudomonadaceae, Pseudomonas Gram - AmR, StR, SpR, RfR
7.  711 (-), Proteobacteria, Oxalobacteriaceae, Janthinobacterium Gram -AmR, SpR
8.  712 (-), Proteobacteria, Comamonadaceae, Variovorax Gram -AmR, KmR, StR, GmR, SpR, RfR
9.  714 (+), Actinobacteria, Micrococcaceae, Micrococcus Gram +
10.  715 (+), Actinobacteria, Microbacteriaceae, Curtobacterium Gram +
11.  724 (+), Actinobacteria, Nocardiaceae, Rhodococcus Gram +
12.  812 (+). Bacillus subtilis SSB2 WT Gram +


# Set working directory #

```{r, message=FALSE, warning=FALSE}
# Set working directory #
rm(list = ls())
getwd()
setwd("C:/Users/Venus/Github/Rpf/generality/")

# Require or install packages #
package.list <- c('vegan', 'ggplot2', 'data.table' , 'nlme' , 'plyr', 'dplyr', 'reshape2', 'stats', 'agricolae') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }
```

## Description of the data set 

growthcurveparameters.csv : A csv file containing the growth curve parameters extracted from 20160320_rawgrowthcurve.csv and 20160323_rawgrowthcurve.csv using the Growth_Curve Repo code. The parameters described are KBS strain, phylum, genus, rep, gram, treatment, b0, A, umax, L, z, umax.lw, umax.up, umax.lw, umax.up.FI. 

gc delta.csv: A csv file containing the proportional differences between treatments for umax, A, and L growth curve parameter based on growth curve output from raw absorbance readings.

```{r}
setwd("C:/Users/Venus/Github/Rpf/generality/")

# Read growth curve parameter output #
gc <- read.csv("growthcurveparameters.csv", header = T)

# Remove NA from data set # 
gc <- na.omit(gc)

# Subset Genus from data set # 
umax.gc <- subset(gc, select = c("Genus", "Treatment", "Rep", "umax"))
A.gc <- subset(gc, select = c("Genus", "Treatment", "Rep", "A"))
L.gc <- subset(gc, select = c("Genus", "Treatment", "Rep", "L"))

# Calculate mean values of umax, A, and L # 
umax.mean <- ddply(umax.gc, c("Genus", "Treatment"), summarise, mean=mean(umax))
A.mean <- ddply(A.gc, c("Genus", "Treatment"), summarise, mean=mean(A))
L.mean <- ddply(L.gc, c("Genus", "Treatment"), summarise, mean=mean(L))


# Load proportional changes for growth curve parameter data set # 
data <- read.csv("gcdelta.csv", header=T)

# Remove NA from data set # 
data <- na.omit(data)

# Make phylogenetic distance a factor # 
data$PDKBS0714 <- factor(data$PDKBS0714)

# Subsetting each growth curve parameters from data set # 
umax <- subset(data,  select = c("Strain", "Phylum", "Genus", "Rep", "Gram", "udelta"))
A <- subset(data, select = c("Strain", "Phylum", "Genus", "Rep", "Gram", "ADelta"))
L <- subset(data, select = c("Strain", "Phylum", "Genus", "Rep", "Gram", "Ldelta"))
```

## 3) Visualizing dataset

Making boxplot figures outlining the proportional changes of growth curve parameters with Rpf treatment

```{r}
# Assigning color to each phylum box plots # 
library(RColorBrewer)
myColors <- brewer.pal(4,"Set1")
names(myColors) <- levels(data$Phylum)
colScale <- scale_colour_manual(name = "Phylum",values = myColors)


# Boxplot showing the difference in umax between treatments for each strain # 
u.boxplot <- ggplot(data, aes(x=PDKBS0714, y = udelta, fill = Phylum)) + 
  geom_boxplot(lwd = 0.75) +
  colScale
  

u.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Maximum growth rate"))) +
  theme_classic() +
  scale_x_discrete(labels=c("Micrococcus","Arthrobacter",
                            "Curtobacterium","Rhodococcus", "Mycobacteriaceae",
                            "Bacillus", "Azospirillium", "Janthinobacterium",
                            "Pseudomonas", "Variovorax", "Pedobacter")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 35, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,5,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(5,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 1.5),
        legend.title = element_text(size=14),
        legend.text=element_text(size=13.5)) +
  geom_hline(yintercept = 0, linetype="dashed", size = 1) + 
  annotate("text", x = 1.1 , y=0.45, label = c("A"), size=10)


# Box plot showing lag time difference between treatments # 
l.boxplot <- ggplot(data, aes(x=PDKBS0714, y = Ldelta,  fill = Phylum)) + 
  geom_boxplot(lwd = 0.75) +
  colScale

l.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Lag time"))) +
  theme_classic() +
  scale_x_discrete(labels=c("Micrococcus","Arthrobacter",
                            "Curtobacterium","Rhodococcus", "Mycobacteriaceae",
                            "Bacillus", "Azospirillium", "Janthinobacterium",
                            "Pseudomonas", "Variovorax", "Pedobacter")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 35, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,5,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(5,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 1.5),
        legend.title = element_text(size=14),
        legend.text=element_text(size=13.5)) +
  geom_hline(yintercept = 0, linetype="dashed", size =1) + 
  annotate("text", x = 1.15 , y=1.85, label = c("B"), size=10)


# Box plot showing Biomass difference between treatments # 
a.boxplot <- ggplot(data, aes(x=PDKBS0714, y = ADelta, fill = Phylum)) + 
  geom_boxplot(lwd = 0.75) +
  colScale

a.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Biomass yield"))) +
  theme_classic() +
    scale_x_discrete(labels=c("Micrococcus","Arthrobacter",
                            "Curtobacterium","Rhodococcus", "Mycobacteriaceae",
                            "Bacillus", "Azospirillium", "Janthinobacterium",
                            "Pseudomonas", "Variovorax", "Pedobacter")) +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 35, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,5,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(5,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 1.5),
        legend.title = element_text(size=14),
        legend.text=element_text(size=13.5)) +
  geom_hline(yintercept = 0, linetype="dashed", size =1) + 
  annotate("text", x = 1.1 , y=0.275, label = c("C"), size=10)

```

## 4) Rpf generality statistical analysis 

I performed Two Factor ANOVAs to test for significant effect of Rpf on growth curves and then post hoc TukeyHSF to determine direction of Rpf effect on each bacterial strains. 

```{r}
# Maximum growth rate # 
# Two factor ANOVA # 
umax.aov <- anova(lm(umax ~ Treatment*Genus, data=umax.gc))
umax.aov
# One factor ANOVA # 
u1 <- aov(umax~Treatment*Genus, data=umax.gc)
# Post hoc TukeyHSD 
posthoc <- TukeyHSD(x=u1, 'Treatment', conf.level=0.95)
posthoc
# Multiple comparison Tukey
hsd.u <- HSD.test(u1, 'Treatment')
hsd.u # Rpf treatment within strain genus has a strong effect on umax (F = 2.7188, p=0.006854) #


# Lag time to exponential growth # 
# Two factor ANOVA # 
L.aov <- anova(lm(L ~ Treatment*Genus, data=L.gc))
L.aov
# One factor ANOVA # 
l1 <- aov(L~Treatment*Genus, data=L.gc)
# Post hoc TukeyHSD 
posthoc <- TukeyHSD(x=l1, 'Treatment', conf.level=0.95)
posthoc
# Multiple comparison Tukey
hsd.l <- HSD.test(l1, 'Treatment')
hsd.l # Interaction between genus and Treatment was significant (p = 0.0009556) #


# Final biomass yeild # 
# Two factor ANOVA # 
A.aov <- anova(lm(A ~ Treatment*Genus, data=A.gc))
A.aov
# One factor ANOVA # 
a1 <- aov(A~Treatment*Genus, data=A.gc)
# Post hoc TukeyHSD 
posthoc <- TukeyHSD(x=a1, 'Treatment', conf.level=0.95)
posthoc
# Multiple comparison Tukey
hsd <- HSD.test(a1, 'Treatment')
hsd

```

## 5) Phylogenetic Analysis 

Generate a phylogenetic tree of KBS study strains and then overlay with bar graph of proportional changes with each growth curve parameter. 

```{r, message=FALSE, warning=FALSE}
# Set working directory 
rm(list = ls())
setwd("C:/Users/Venus/Github/Rpf/generality/")

# Load packages # 
package.list <- c('muscle', 'ape', 'seqinr', 'phangorn', 'phylobase', 'adephylo', 'geiger', 'picante', 'stats', 'RColorBrewer', 'caper')
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }
```

## Load data set 

KBSsequences.fasta : A fasta file containing the 16s RNA sequences from each of the KBS strain studied.

```{r}
# Read FASTA file as DNAStringSet #
dna <- readDNAStringSet("KBSsequences.fasta")
```

# Aligning KBS study sequences using Ape package # 

```{r}
# Performing an Alignment using Muscle from Ape package #
aln <- muscle::muscle(seqs="dna.fasta", stringset = dna, out = "dna.afa")

# Convert Alignment File to DNAbin Object #
dna.bin <- as.DNAbin(aln)
dna.bin.1 <- as.phyDat(aln)

# Create Distance Matrix with "raw" Model {ape}
seq.dist.raw <- dist.dna(dna.bin, model = "raw", pairwise.deletion = FALSE)
```

# Constructing neighbor joining tree #

```{r}
# Neighbor Joining Algorithm to Construct Tree, a 'phylo' Object {ape}
nj.tree <- bionj(seq.dist.raw)

# Maximum Likelihood Algorithm to Construct Tree
#fit <- pml(nj.tree, data=dna.bin.1)

# Identify Outgroup Sequence
outgroup <- match("Methanosarcina", nj.tree$tip.label)

# Root the Tree {ape}
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)

# Plot the Rooted Tree{ape}
par(mar = c(1,1,2,1) + 0.1)
plot.phylo(nj.rooted, main = "Neigbor Joining Tree", "phylogram", use.edge.length = FALSE,
           direction = "right", cex = 1, label.offset = 1)
add.scale.bar(cex = 0.7)
```

# Creating distance matrix with "F84" model in Ape # 

```{r}
# Create distance matrix with "F84" model {ape}
seq.dist.F84 <- dist.dna(dna.bin, model = "F84", pairwise.deletion = FALSE)

# Plot Distances from Different DNA Substitution Models
par(mar = c(5, 5, 2, 1) + 0.1)
plot(seq.dist.raw, seq.dist.F84,
     pch = 20, col = "red", las = 1, asp = 1, xlim = c(0, 0.7), ylim = c(0, 0.7),
     xlab = "Raw Distance", ylab = "F84 Distance")
abline(b = 1, a = 0, lty = 2)
text(0.65, 0.6, "1:1")

# Make Neighbor Joining Trees Using Different DNA Substitution Models {ape}
raw.tree <- bionj(seq.dist.raw)
F84.tree <- bionj(seq.dist.F84)

# Define Outgroups
raw.outgroup <- match("Methanosarcina", raw.tree$tip.label)
F84.outgroup <- match("Methanosarcina", F84.tree$tip.label)

# Root the Trees {ape}
raw.rooted <- root(raw.tree, raw.outgroup, resolve.root=TRUE)
F84.rooted <- root(F84.tree, F84.outgroup, resolve.root=TRUE)

# Make Cophylogenetic Plot {ape}
layout(matrix(c(1,2), 1, 2), width = c(1, 1))
par(mar = c(1, 1, 2, 0))
plot.phylo(raw.rooted, type = "phylogram", direction = "right", show.tip.label=TRUE,
           use.edge.length = FALSE, adj = 0.5, cex = 0.6, label.offset = 2, main = "Raw")
par(mar = c(1, 0, 2, 1))
plot.phylo(F84.rooted, type = "phylogram", direction = "left", show.tip.label=TRUE,
           use.edge.length = FALSE, adj = 0.5, cex = 0.6, label.offset = 2, main = "F84") # Perfect Match #
```

# Integrating KBS growth curve changes with Phylogeny # 

```{r}
# Import Growth Rate Data
p.growth <- read.table("averagedgrowthcurve.txt", sep = "\t", header = TRUE, row.names = 1)

# Generating a tree with traits #
# Generate Neighbor Joining Tree Using F84 DNA Substitution Model {ape}
nj.tree <- bionj(seq.dist.F84)

# Define the Outgroup
outgroup <- match("Methanosarcina", nj.tree$tip.label)

# Create a Rooted Tree {ape}
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)

# Keep Rooted but Drop Outgroup Branch
nj.rooted <- drop.tip(nj.rooted, "Methanosarcina")

# Define Color Palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"))

# Map Growth curve parameter response to Rpf treatment {adephylo}
par(mar=c(1,1,1,1) + 0.1)
x <- phylo4d(nj.rooted, p.growth)
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE,
              cex.label = 0.8, scale = TRUE, use.edge.length = FALSE,
              edge.color = "black", edge.width = 2, box = FALSE,
              col=mypalette(25), pch = 15, cex.symbol = 2.5,
              ratio.tree = 0.6, cex.legend = 1, center = FALSE)
```

# Visualize Trees With Different Levels of Phylogenetic Signal {geiger} # 
# Finding the Phylogenetic Signal #

```{r}
nj.lambda.5 <- rescale(nj.rooted, "lambda", 0.5)
nj.lambda.0 <- rescale(nj.rooted, "lambda", 0)
layout(matrix(c(1,2,3), 1, 3), width = c(1, 1, 1))
par(mar=c(1,0.5,2,0.5)+0.1)
plot(nj.rooted, main = "lambda = 1", cex = 0.7, adj = 0.5)
plot(nj.lambda.5, main = "lamba = 0.5", cex = 0.7, adj = 0.5)
plot(nj.lambda.0, main = "lamba = 0", cex = 0.7, adj = 0.5)

# Generate Test Statistics for Comparing Phylogenetic Signal {geiger}
fitContinuous(nj.rooted, p.growth, model = "lambda")
fitContinuous(nj.lambda.0, p.growth, model = "lambda")

# There is a phylogenetic signal of rpf responsiveness, lambda=1 and lambda=0 is AIC is the same #
```

## 6) Rpf generality figure based on phylogenetic tree construction 

A trait based tree for 1st Rpf figure

```{r, message=FALSE, warning=FALSE}
# Set working directory # 
rm(list = ls())
setwd("C:/Users/Venus/Github/Rpf/generality/")
# Load required packages # 
package.list <- c('ape', 'seqinr', 'phylobase', 'adephylo', 'geiger', 'picante', 'stats', 'RColorBrewer', 'caper', 'phylolm', 'pmc', 'ggplot2', 'tidyr', 'dplyr', 'phangorn', 'pander', 'phytools') 
for (package in package.list) {
  if (!require(package, character.only=TRUE, quietly=TRUE)) {
    install.packages(package)
    library(package, character.only=TRUE)
  }
}
```

# Create trait database # 

```{r}
# Load growth curve traits # 

# Standardize traits (mean=0, std = 1)
traits <- data.frame(m2[,-1])
rownames(traits) <- m2[,1] 
traits.norm <- scale(traits)

# Check scaling
check.norm.mean <- round(colMeans(traits.norm, na.rm = TRUE), 1)
check.norm.sd <- apply(traits.norm, 2, sd, na.rm = TRUE)


```

# Make tree using RAxML has already been done and contains all KBS study strains frm Dimensions/phylo --> I may have to go back and remove other strains

# Map traits onto tree 

```{r}
# Load ML tree
ml.tree <- read.tree("20170621_Tree//RAxML_bipartitionsBranchLabels.LTDE.ML")

# Define the outgroup
outgroup <- match("Methanosarcina", ml.tree$tip.label)

# Create a rooted tree {ape}
ml.rooted <- root(ml.tree, outgroup, resolve.root = TRUE)

# Keep rooted but drop outgroup branch
ml.rooted <- drop.tip(ml.rooted, "Methanosarcina")

# Define color palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"))

# Function to remove NAs from trait database *:
# *https://cran.r-project.org/web/packages/adephylo/adephylo.pdf
f1 <- function(vec){
  if(any(is.na(vec))){
  m <- mean(vec, na.rm=TRUE)
  vec[is.na(vec)] <- m
}
  return(vec)
}

# Normalized traits with na.omit
traits.norm.na <- f1(traits.norm)

# Map traits {adephylo}
par(mar=c(1,1,1,1) + 0.1)
x <- phylo4d(ml.rooted, traits.norm.na)
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE, 
              cex.label = 0.75, scale = FALSE, use.edge.length = FALSE, 
              edge.color = "black", edge.width = 2, box = FALSE, 
              col = mypalette(25), pch = 15, cex.symbol = 4.5, 
              ratio.tree = 0.5, cex.legend = 1.5, center = FALSE)  

```








