---
title: "Untitled"
author: "Venus"
date: "April 11, 2017"
output: html_document
---
# Set up # 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# Set working directory #
rm(list = ls())
getwd()
setwd("C:/Users/Venus/Github/Rpf/generality/growthparameterchanges/")

# Require or install packages #
package.list <- c('vegan', 'ggplot2', 'data.table' , 'nlme' , 'plyr', 'dplyr', 'reshape2', 'stats') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
  } }
```

# Data visualization #

```{r}
setwd("C:/Users/Venus/Github/Rpf/generality/growthparameterchanges/")
# I manually calculated and input proportion (Rpf-control/control) into csv files # 
dat <- read.csv("gcdelta.csv", header=T)

# Boxplot showing the difference in umax between treatments for each strain # 
u.boxplot <- ggplot(dat, aes(x=Genus, y = udelta)) + 
  geom_boxplot(lwd = 0.75) +
  theme_bw()

u.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Maximum growth rate"))) +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 45, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,15,0,0)),
        axis.title.x = element_text(colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  geom_hline(yintercept = 0, linetype="dotted")


# Box plot showing lag time difference between treatments # 
l.boxplot <- ggplot(dat, aes(x=Genus, y = Ldelta)) + 
  geom_boxplot(lwd = 0.75) +
  theme_bw()

l.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Lag time"))) +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 45, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,0)),
        axis.title.x = element_text(colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  geom_hline(yintercept = 0, linetype="dotted")


# Box plot showing Biomass difference between treatments # 
a.boxplot <- ggplot(dat, aes(x=Genus, y = ADelta)) + 
  geom_boxplot(lwd = 0.75) +
  theme_bw()

a.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Biomass yield"))) +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 45, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,0)),
        axis.title.x = element_text(colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  geom_hline(yintercept = 0, linetype="dotted")


```

# Phylogenetic Distance to KBS0714 #
# Load and visualize dataset # 

```{r}
rm(list = ls())
# Set working directory # 
setwd("C:/Users/Venus/Github/Rpf/generality/")
# Load packages # 
package.list <- c('muscle', 'ape', 'seqinr', 'phylobase', 'adephylo', 'geiger', 'picante', 'stats', 'RColorBrewer', 'caper', 'phylolm', 'pmc', 'ggplot2', 'tidyr', 'dplyr', 'phangorn', 'pander') 
for (package in package.list) {
  if (!require(package, character.only=T, quietly=T)) { 
    install.packages(package)
    library(package, character.only=T)
} }

# Read FASTA file as DNAStringSet #
dna <- readDNAStringSet("KBSsequences.fasta")

# Performing an Alignment using Muscle from Ape package #
aln <- muscle::muscle(seqs="dna.fasta", stringset = dna, out = "dna.afa")

# Convert Alignment File to DNAbin Object #
dna.bin <- as.DNAbin(aln)
dna.bin.1 <- as.phyDat(aln)

# Identify Base Pair Region of 16S rRNA Gene to Visualize #
window <- dna.bin[, 100:500]

# Command to visualize sequence alignment #
image.DNAbin(window, cex.lab = 0.50)

# Adds grid to help visualize rows of sequences #
grid(ncol(window), nrow(window), col="lightgrey")


```


# Making phylogenetic tree and distance matrix from KBS0714 # 

```{r}
# Create Distance Matrix with "raw" Model {ape}
seq.dist.raw <- dist.dna(dna.bin, model = "raw", pairwise.deletion = FALSE)

# Neighbor Joining Algorithm to Construct Tree, a 'phylo' Object {ape}
nj.tree <- bionj(seq.dist.raw)

# Identify Outgroup Sequence
outgroup <- match("Methanosarcina", nj.tree$tip.label)

# Root the Tree {ape}
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)

# Plot the Rooted Tree{ape}
par(mar = c(1,1,2,1) + 0.1)
plot.phylo(nj.rooted, main = "Neigbor Joining Tree", "phylogram", use.edge.length = FALSE,
           direction = "right", cex = 1, label.offset = 1)
add.scale.bar(cex = 0.7)


```

# Maximum liklihood tree # 

```{r}
# Requires alignment to be read in with as phyDat object #
fit <- pml(nj.rooted, data=dna.bin.1)
# Fit tree using a JC69 substitution model # 
fitJC <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE)
# Model selection with either an ANOVA test or by AIC value #
anova(fitJC, fitGTR)
AIC(fitJC)
AIC(fitGTR) # Takes a long time... # 

```

```{r}
# Creating patristic distance between all taxa # 
pd.matrix <- cophenetic(nj.tree)

# Patristic distance of all taxa to KBS0714
KBS0714.pd <- as.data.frame(pd.matrix[9, ])
colnames(KBS0714.pd) <- c("Actinobacteria, Micrococcus")

# Sorted the patristic distance by hand in excel # 
pd <- read.csv("PDKBS0714.csv", header=T) # Inputed the distance values into gcdelta # 

```

# PD figure #
# Scatter plots # 

```{r}
rm(list = ls())
setwd("C:/Users/Venus/Github/Rpf/generality/growthparameterchanges/")
# I manually calculated and input proportion (Rpf-control/control) into csv files # 
dat <- read.csv("gcdelta.csv", header=T)

dat$PDKBS0714 <- as.factor(dat$PDKBS0714)

# Boxplot showing the difference in umax between treatments for each strain # 
u.scatter <- ggplot(dat, aes(x=PDKBS0714, y = udelta)) + 
  geom_point(size = 2) +
  theme_bw()

u.scatter + 
  labs(x="Genera", y=expression(paste(Delta ,"Maximum growth rate"))) +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 45, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,15,0,0)),
        axis.title.x = element_text(colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  geom_hline(yintercept = 0, linetype="dotted") 

  geom_hline(slope = pd.u)
pd.u <- lm(udelta ~ PDKBS0714, data = dat)




```





```{r}

# Box plot showing lag time difference between treatments # 
l.scatter <- ggplot(dat, aes(x=PDKBS0714, y = Ldelta)) + 
  geom_boxplot(lwd = 0.75) +
  theme_bw()

l.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Lag time"))) +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 45, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,0)),
        axis.title.x = element_text(colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  geom_hline(yintercept = 0, linetype="dotted")


# Box plot showing Biomass difference between treatments # 
a.boxplot <- ggplot(dat, aes(x=PDKBS0714, y = ADelta)) + 
  geom_boxplot(lwd = 0.75) +
  theme_bw()

a.boxplot + 
  labs(x="Genera", y=expression(paste(Delta ,"Biomass yield"))) +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size=18),
        axis.text.x=element_text(colour="black", size =12, angle = 45, hjust = 1),
        axis.title=element_text(size=20),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,0)),
        axis.title.x = element_text(colour="black",margin=margin(15,0,0,0)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=2, fill=NA)) +
  geom_hline(yintercept = 0, linetype="dotted")

```





