---
title: "Rpf"
author: "Jay T. Lennon and Venus Kuo"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
output: pdf_document
geometry: margin=2.54cm
---

Termination of dormancy in soil bacteria with resuscitation promoting factor (Rpf)

```{r setup}
rm(list=ls())
getwd()
#setwd("~/GitHub/Rpf")
knitr::opts_knit$set(root.dir = '~/../GitHub/Rpf') # For Venus's computer
```

## Load packages and functions

```{r}
require("png")
require("bbmle")
require("drc")
require("ggplot2")
require("grid")
require("plyr")

sem <- function(x) sqrt(var(x)/length(x))
```

## 1) Enzymatic activity of recombinant Rpf

```{r}
enz <- read.table("data/enzyme.activity/brent.txt", sep="\t", header=TRUE)

# Starting values for fitting Michaelis-Menten with maximum likelihood
V = 85000 
K = 0.025 
Z = 100

fit <- mle2(enz$abs ~ dnorm(mean = v * enz$sub / (k + enz$sub), sd = z), 
    start = list(v = V, k = K, z = Z), data = enz)

# Plot data
#png(filename="~/GitHub/Rpf/figures/Fig1.png",
#    width = 1200, height = 1200, res = 96*2)

png(filename="~/../GitHub/Rpf/figures/Fig1.png", # For Venus
    width = 1200, height = 1200, res = 96*2) 

plot.new()
par(mar = c(7, 7, 5, 7))

plot(enz[,1], enz[,2], xlim = c(-0.05, 0.3), 
     ylim = c(-5000, 90000), type = "p", 
     pch = 22, bg = "grey", col = "black", cex = 2, ylab = "", xlab = "", 
     cex.lab = 1.5, las = 1, lwd = 2, yaxt = "n", xaxt = "n")
box(lwd=2)

# Add ticks and tick labels
axis(side = 2, lwd.ticks = 2, las = 1, cex.axis = 1.25, 
   labels = c(0, 20000, 40000, 60000, 80000), at = c(0, 20000, 40000, 60000, 80000))

axis(side = 4, labels = F, lwd.ticks = 2, 
   at = c(0, 20000, 40000, 60000, 80000))

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1, mgp = c(3, 1, 0),
    labels = c(0, 0.10, 0.2, 0.3), at = c(0, 0.10, 0.2, 0.3))

axis(side = 3, labels = F, lwd.ticks = 2, las = 1, cex.axis = 1.25, 
   at = c(0, 0.10, 0.2, 0.3))

mtext('Peptidoglycan (mg)', side = 1, outer = TRUE, cex = 1.5, 
      line = -4, adj = 0.5)

mtext('Relative Fluorescence', side = 2, outer = TRUE, cex = 1.5, line = -2, adj = 0.6)

# Plot function
curve((coef(fit)[[1]] * x) / (coef(fit)[[2]] + x), from = 0, to = 0.25, add = TRUE, lty = 2, lwd = 2.5)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("figures/Fig1.png")
grid.raster(img)
```

## 2) Growth of KBS0714 in response to recombinant Rpf

```{r}
growth <- read.table("data/growth.curve/growth.curve.txt", sep="\t", header=TRUE)
head(growth)
#growth.early <- growth.raw[1:29,]
#growth.late <- growth.raw[29:nrow(growth.raw),1:5]

# Plot data
#png(filename="~/GitHub/Rpf/figures/Fig2.png", 
#    width = 1200, height = 900, res = 96*2)

png(filename="~/../GitHub/Rpf/figures/Fig2.png",  # Venus's computer
    width = 1200, height = 900, res = 96*2)

plot.new()
par(mar = c(7, 7, 5, 7))

# Add +Rpf tubes
plot(growth[,1], growth[,6], xlim = c(-25, 625), ylim = c(-0.5, 4), type = "l", 
     lty = 2, col = "black", lwd = 2, ylab = "", xlab = "", 
     cex.lab = 1.5, las = 1, yaxt = "n", xaxt = "n")
box(lwd = 2)

points(growth[,1], growth[,7], type = "l", lty = 2, col = "black", lwd = 2)
points(growth[,1], growth[,8], type = "l", lty = 2, col = "black", lwd = 2)
points(growth[,1], growth[,9], type = "l", lty = 2, col = "black", lwd = 2)

# Add -Rpf tubes
points(growth[,1], growth[,2], type = "l", lty = 3, col = "black", lwd = 2)
points(growth[,1], growth[,3], type = "l", lty = 3, col = "black", lwd = 2)
points(growth[,1], growth[,4], type = "l", lty = 3, col = "black", lwd = 2)
points(growth[,1], growth[,5], type = "l", lty = 3, col = "black", lwd = 2)

# Add ticks and tick labels
axis(side = 2, lwd.ticks = 2, las = 1, cex.axis = 1.25, 
   labels = c("0", "1", "2", "3", "4"), 
        at = c(0, 1, 2, 3, 4))

axis(side = 4, labels = F, lwd.ticks = 2, 
   at = c(0, 1, 2, 3, 4))

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1, mgp = c(1, 1, 0),
    labels = c(0, 200, 400, 600), at = c(0, 200, 400, 600))

axis(side = 3, labels = F, lwd.ticks = 2, las = 1, cex.axis = 1, 
   at = c(0, 200, 400, 600))

mtext('Biomass (OD600)', side = 2, outer = TRUE, cex = 1.5, 
      line = -4.5, adj = 0.6)

mtext('Time (h)', side = 1, outer = TRUE, cex = 1.5, 
      line = -4, adj = 0.5)

text(400, 3.6, labels = "+Rpf", cex = 1)
text(540, 3.6, labels = "-Rpf", cex = 1)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("figures/Fig2.png")
grid.raster(img)

# Growth curve statistics: lag time, umax, and yield

# 1) Need to make plots to check that fits are reasonable 
            #--> plots of fits are in GitHub Repo and Box

# 2) Need to collect paramaters (means +/- sem)
# Load output of modified Gompertz model 
growth.p <- read.delim("data/growth.curve/parameters.txt",  sep=",", header=TRUE)

# Add treatments (A1-4 = Rpf-, B1-4 = Rpf+)
Treatment <- c("Rpf-", "Rpf-","Rpf-","Rpf-", "Rpf+","Rpf+","Rpf+","Rpf+")
growth.p$Treatment <- Treatment

# Calculate mean and sem for umax
growth.umax.means.sem <- ddply(growth.p, c("Treatment"), summarise,
                   mean=mean(umax), sem=sd(umax)/sqrt(length(umax)))
growth.umax.means.sem <- transform(growth.umax.means.sem, lower=mean-sem, upper=mean+sem)
# Calculate mean and sem for lag time
growth.lag.means.sem <- ddply(growth.p, c("Treatment"), summarise,
                   mean=mean(L), sem=sd(L)/sqrt(length(L)))
growth.lag.means.sem <- transform(growth.lag.means.sem, lower=mean-sem, upper=mean+sem)
# Calculate mean and sem for biomass yield 
growth.A.means.sem <- ddply(growth.p, c("Treatment"), summarise,
                   mean=mean(A), sem=sd(A)/sqrt(length(A)))
growth.A.means.sem <- transform(growth.A.means.sem, lower=mean-sem, upper=mean+sem)

# 3) Need to statistically test (t-test) effect of Rpf
# Maximum growth rate
growth.umax.t <- t.test(umax~Treatment, data=growth.p)
growth.umax.t # Non-significant effect of Rpf on umax p = 0.28
# Lag time
growth.L.t <- t.test(L~Treatment, data=growth.p)
growth.L.t # Significant effect of Rpf on L p = 0.0069
# Biomass yeild
growth.A.t <- t.test(A~Treatment, data=growth.p)
growth.A.t # Non-significant effect of Rpf on A p = 0.34
```

## 3) Antibody control

```{r}
antibody <- read.table("data/antibody/antibody.txt", header = TRUE, sep="\t")

# Table
rpf.mean <- aggregate(antibody$OD600 ~ Treatment, mean, data = antibody)
rpf.sem <- aggregate(antibody$OD600 ~ Treatment, sem, data = antibody)
LL.95 <- aggregate(antibody$OD600 ~ Treatment, antibody,
          FUN = function(x) t.test(x)$conf.int[1])
UL.95 <- aggregate(antibody$OD600 ~ Treatment, antibody,
          FUN = function(x) t.test(x)$conf.int[2])
table <- data.frame(rpf.mean[1], rpf.mean[2], rpf.sem[2],
          LL.95[2], UL.95[2])
colnames(table) <- c("Treatment", "mean", "sem", "LCI", "UCI")

# ANOVA
antibody.anova <- aov(OD600 ~ Treatment, data = antibody)
summary(antibody.anova)
  # reject null; P <0.0001
TukeyHSD(antibody.anova)
  # no difference between -Rpf and +Rpf,+anti.28
  # no difference between +Rpf and +Rpf,+anti.0

# Subset data
no.rpf <- antibody[ which(antibody$Treatment == "Rpf-"),]
plus.rpf <- antibody[ which(antibody$Treatment == "Rpf+"),]
plus.rpf.anti.0 <- antibody[ which(antibody$Treatment == "Rpf+, Anti0+"),]
plus.rpf.anti.28 <- antibody[ which(antibody$Treatment == "Rpf+, Anti28+"),]

# Plot data
#png(filename="~/GitHub/Rpf/figures/Fig3.png",
#    width = 1200, height = 1200, res = 96*2)

png(filename="~/../GitHub/Rpf/figures/Fig3.png",  # Venus's Computer
    width = 1200, height = 1200, res = 96*2)

plot.new()
par(mar = c(7, 7, 5, 7))

antibody.point <- plot(jitter(rep(1, length(no.rpf$OD600)), amount = 0.1), no.rpf$OD600, 
      ylim = c(-0.2, 3.5), xlim = c(0.5, 5.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 2,
      cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 1.5,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
points(jitter(rep(2.5, length(plus.rpf$OD600)), amount = 0.1), plus.rpf$OD600, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)
points(jitter(rep(3.5, length(plus.rpf.anti.0$OD600)), amount = 0.1), plus.rpf.anti.0$OD600, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)
points(jitter(rep(4.5, length(plus.rpf.anti.28$OD600)), amount = 0.1), plus.rpf.anti.28$OD600, pch = 21, 
       bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

points(1, mean(no.rpf$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(2.5, mean(plus.rpf$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  
points(3.5, mean(plus.rpf.anti.0$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(4.5, mean(plus.rpf.anti.28$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 

mtext(expression('Biomass (OD600)'), side = 2,
      outer = FALSE, cex = 1.5, line = 3.6, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.0", "1.0", "2.0", "3.0"), at = c(0, 1, 2, 3))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(0, 1, 2, 3), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(1, 2.5, 3.5, 4.5), labels = F)

arrows(x0 = c(1, 2.5, 3.5, 4.5), y0 = table$mean, y1 = table$LCI, angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(1, 2.5, 3.5, 4.5), y0 = table$mean, y1 = table$UCI, angle = 90,
       length=0.1, lwd = 2)

box(lwd = 2)

# Make x-axis labels
mtext('-Rpf', side = 1, line = 4.5, at = 1, cex = 1.5)
mtext('+Rpf', side = 1, line = 4.5, at = 3.5, cex = 1.5)
mtext('-', side = 1, line = 0.75, at = 2.5, cex = 2)
mtext('-', side = 1, line = 2.25, at = 2.5, cex = 2)
mtext('+', side = 1, line = 0.75, at = 3.5, cex = 2)
mtext('-', side = 1, line = 2.25, at = 3.5, cex = 2)
mtext('+', side = 1, line = 0.75, at = 4.5, cex = 2)
mtext('+', side = 1, line = 2.25, at = 4.5, cex = 2)
mtext('Antibody', side = 1, line = 0.75, at = 5.8, cex = 1.5)
mtext('Antigen', side = 1, line = 2.25, at = 5.7, cex = 1.5)

segments(0.7, -1.1, 1.5, -1.1, col = "black", lwd = 2, xpd = TRUE)
segments(2.25, -1.1, 4.75, -1.1, col = "black", lwd = 2, xpd = TRUE)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("figures/Fig3.png")
grid.raster(img)
```

## 4) Dose-response curve of Rpf

```{r}
# load data
dose.raw <- read.table("data/dose.response/dose.response.txt", header = TRUE, sep="\t")

# calculate area of plate with bacteri
area <- dose.raw$Pixels / (pi * dose.raw$Plate.Width * dose.raw$Plate.Height)
dose.raw <- data.frame(dose.raw, area)

# select 48 or 96 hrs
biomass <- dose.raw[which(dose.raw$Hrs == 48),]

# calculate total biomass across plates
sum.area <- sum(biomass$area)

# calcualte relative biomass on each plate with different Rpf concentrations
rel.biomass <- biomass$area/sum.area
dose <- data.frame(biomass, rel.biomass)

# Starting values for fitting asymmetric Laplace with maximum likelihood
A = 0.40 # Maximum biomass (15000)
X = 45 # Optimum Rpf (55)
K = 0.5 # Kappa; <1 right side decays slowly; >1 left side decays slowly (5)
L = -0.5 # Lamba; peakiness (0.1); larger = flatter
Z = 10 # Error (5)

fit <- mle2(rel.biomass ~ dnorm(mean = a*exp(-(Rpf-x) * l * sign(x - Rpf)* 
            k ^sign(x - Rpf)), sd = z), start = list(a = A, x = X, k = K, l = L, z = Z), 
            data = dose)

# Plot data
#png(filename="~/GitHub/Rpf/figures/Fig4.png",
#    width = 1200, height = 1200, res = 96*2)

png(filename="~/../GitHub/Rpf/figures/Fig5.png",  # Just Venus's computer
    width = 1200, height = 1200, res = 96*2)

plot.new()
par(mar = c(7, 7, 5, 7))

plot(dose[,2], dose[,8], xlim = c(-25, 185), 
     ylim = c(0, 0.5), type = "p", 
     pch = 22, bg = "grey", col = "black", cex = 2, ylab = "", xlab = "", 
     cex.lab = 1.5, las = 1, lwd = 2, yaxt = "n", xaxt = "n")
box(lwd=2)

# Add ticks and tick labels
axis(side = 2, lwd.ticks = 2, las = 1, cex.axis = 1.25, 
   labels = c(0, 0.25, 0.5), at = c(0, 0.25, 0.5))

axis(side = 4, labels = F, lwd.ticks = 2, 
   at = c(0, 0.25, 0.5))

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1, mgp = c(3, 1, 0),
    labels = c(0, 50, 100, 150), at = c(0, 50, 100, 150))

axis(side = 3, labels = F, lwd.ticks = 2, las = 1, cex.axis = 1.25, 
   at = c(0, 50, 100, 150))

mtext('Rpf (nM)', side = 1, outer = TRUE, cex = 1.5, 
      line = -4, adj = 0.5)

mtext(expression(paste('Relative Biomass')), 
      side = 2, outer = TRUE, cex = 1.5, line = -3, adj = 0.6)

# Plot function
curve(coef(fit)[[1]] * exp(-(x-coef(fit)[[2]]) * coef(fit)[[4]] * sign(coef(fit)[[2]] - x) * 
      coef(fit)[[3]]^sign(coef(fit)[[2]]-x)),from = 0, to = 160, add = TRUE, lty = 2, lwd = 2.5)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("figures/Fig5.png")
grid.raster(img)
```


## 5) Rpf structural diversity

```{r}
###################################################################################
# Commented out analysis of gompertz output #

# Load mutant Rpf Gompertz growth curve parameters
# Site directed mutations of glutamate to alanine, lysine, and glutamine 
#gc.E54A <- read.delim("data/mutant/Rpf_E54A_gc_parameters.txt", sep = ",", header = TRUE)
#gc.E54K <- read.delim("data/mutant/Rpf_E54K_gc_parameters.txt", sep = ",", header = TRUE)
#gc.E54Q <- read.delim("data/mutant/Rpf_E54Q_gc_parameters.txt", sep = ",", header = TRUE)
# Truncated repeating Alanines and LysM domain
#gc.trnc <- read.delim("data/mutant/Rpf_trnc_gc_parameters.txt", sep = ",", header = TRUE)

# Append treatment information to growth curve output 
#gc.mutant.key <- read.delim("data/mutant/mutant.key.txt", sep = "\t", header = TRUE)
#gc.trnc$Treatment <- gc.mutant.key$Treatment[which(gc.mutant.key$Experiment == "trnc")]
#gc.E54A$Treatment <- gc.mutant.key$Treatment[which(gc.mutant.key$Experiment == "E54A")] #Values looked good
#gc.E54K$Treatment <- gc.mutant.key$Treatment[which(gc.mutant.key$Experiment == "E54K")]
#gc.E54Q$Treatment <- gc.mutant.key$Treatment[which(gc.mutant.key$Experiment == "E54Q")]

# Look at the data to see if values are reasonable
#gc.E54K # Werid samples: A5,A6,A7,A10
#gc.E54Q # Weird samples: A5,A6,A7,A8, A12 - The entire E54Q treatment 
#gc.trnc # Weird samples: A1

# Remove odd data points 
#gc.E54K <- gc.E54K[c(-5,-10) ,]
#gc.trnc <- gc.trnc[-1, ]

# One-factor ANOVA on growth curve parameters # 
#E54A.A.anova <- aov(A ~ Treatment, data = gc.E54A)    # Biomass Yeild
#summary(E54A.A.anova)   # Significant effect of Rpf (p=0.0272)*
#E54A.u.anova <- aov(umax ~ Treatment, data = gc.E54A) # Maximum growth rate
#summary(E54A.u.anova)   # Non-significant effect of Rpf (p=0.753)
#E54A.l.anova <- aov(L ~ Treatment, data = gc.E54A)    # Lag time
#summary(E54A.l.anova)   # Highly significant effect of Rpf (p < 0.0001)

#E54K.A.anova <- aov(A ~ Treatment, data = gc.E54K)    # Biomass Yeild
#summary(E54K.A.anova)   # Non-significant effect (p=0.182)
#E54K.u.anova <- aov(umax ~ Treatment, data = gc.E54K) # Maximum growth rate
#summary(E54K.u.anova)   # Significant effect (p=0.00362)
#E54K.l.anova <- aov(L ~ Treatment, data = gc.E54K)    # Lag time
#summary(E54K.l.anova)   # Marginally non-significant effect (p = 0.0636)

#E54Q.A.anova <- aov(A ~ Treatment, data = gc.E54Q)    # Biomass Yeild
#summary(E54Q.A.anova)   
#E54Q.u.anova <- aov(umax ~ Treatment, data = gc.E54Q) # Maximum growth rate
#summary(E54Q.u.anova)   
#E54Q.l.anova <- aov(L ~ Treatment, data = gc.E54Q)    # Lag time
#summary(E54Q.l.anova)   

#trnc.A.anova <- aov(A ~ Treatment, data = gc.trnc)    # Biomass Yeild
#summary(trnc.A.anova)   # Significant effect (p=0.011)*
#trnc.u.anova <- aov(umax ~ Treatment, data = gc.trnc) # Maximum growth rate
#summary(trnc.u.anova)   # Significant effect (p=0.003)
#trnc.l.anova <- aov(L ~ Treatment, data = gc.trnc)    # Lag time
#summary(trnc.l.anova)   # Significant effect (p = 0.00184)
    # Reject null hypothesis --> Mutating gluamate region alters Rpf resuscitative effects
############################################################################################
```



```{r}
# A) Effect of Rpf structural diversity on final OD600 biomass #
# Import final OD600 dataset 
od <- read.delim("data/mutant/mutant.growth.txt", sep="\t", header = TRUE)

# Remove all LMM treatment data 
toBeRemoved <- which(od$Treatment=="LMM")
od<-od[-toBeRemoved,]

# Subset biomass dataset by experiment
od.E54A <- subset(od, Experiment=="E54A")
od.E54K <- subset(od, Experiment=="E54K")
od.E54Q <- subset(od, Experiment=="E54Q")
od.trnc <- subset(od, Experiment=="trnc")

# T-tests statistics of mutation/ truncation on final OD600 
od.E54A.t <- t.test(OD600~Treatment, data=od.E54A) 
od.E54A.t # Significant effect of E54A mutation (p=0.001) --> E54A decreased final od
od.E54K.t <- t.test(OD600~Treatment, data=od.E54K) 
od.E54K.t # Significant effect of E54K mutation (p=0.0002) --> E54K decreased final od
od.E54Q.t <- t.test(OD600~Treatment, data=od.E54Q) 
od.E54Q.t # Significant effect of E54Q mutation (p=0.01) --> E54Q decreased final od
    # Reject null hypothesis --> Altering Rpf active site decreased resuscitative effects
od.trnc.t <- t.test(OD600~Treatment, data=od.trnc) 
od.trnc.t # Significant effect of truncation (p=0.002) --> Trnc decreased final od
    # Reject null hypothesis --> Removing Rpf repeating alaine and LysM domain decreased resuscitative effects

# Calculate SEM and CI table for each protein and experiment in dataset
od.mean <- aggregate(OD600 ~ Treatment, mean, data = od)
od.sem <- aggregate(OD600 ~ Treatment, sem, data = od)
od.LL.95 <- aggregate(od$OD600 ~ Treatment, od,
          FUN = function(x) t.test(x)$conf.int[1])
od.UL.95 <- aggregate(od$OD600 ~ Treatment, od,
          FUN = function(x) t.test(x)$conf.int[2])
od.table <- data.frame(od.mean[1], od.mean[2], od.sem[2],
          od.LL.95[2], od.UL.95[2])
colnames(od.table) <- c("Treatment", "mean", "sem", "LCI", "UCI")

# Subset data by treatment for plotting data
od.E54A.wt <- od[ which(od$Treatment == "E54AWT"), ]
od.E54K.wt <- od[ which(od$Treatment == "E54KWT"), ]
od.E54Q.wt <- od[ which(od$Treatment == "E54QWT"), ]
od.Trnc.wt <- od[ which(od$Treatment == "TrncWT"), ]
od.E54A <- od[ which(od$Treatment == "E54A"), ]
od.E54K <- od[ which(od$Treatment == "E54K"), ]
od.E54Q <- od[ which(od$Treatment == "E54Q"), ]
od.trnc <- od[ which(od$Treatment == "Trnc"), ]


# B) Effect of Rpf structural diversity on enzymatic activity #
# Load enz check dataset 
enz <- read.delim("data/mutant/mutant.enzyme.txt", sep = "\t", header = TRUE) 

# Subset dataset by experiment for analysis 
enz.E54A <- subset(enz, Experiment == "E54A")
enz.E54K <- subset(enz, Experiment == "E54K")
enz.E54Q <- subset(enz, Experiment == "E54Q")
enz.trnc <- subset(enz, Experiment == "trnc")

# Statistics on EnzCheck results
enz.E54A.t <- t.test(Absorbance~Protein, data=enz.E54A) 
enz.E54A.t # Non-significant effect (p=0.1107) 
enz.E54K.t <- t.test(Absorbance~Protein, data=enz.E54K) 
enz.E54K.t # Non-signficant effect (p=0.1937) 
enz.E54Q.t <- t.test(Absorbance~Protein, data=enz.E54Q) 
enz.E54Q.t # Marginally non-significant effect (p=0.05599) 
#enz.trnc.t <- t.test(Absorbance~Protein, data=enz.trnc) # In the process of redoing this assay
#enz.trnc.t

# Subset data by protein type for figure
enz.E54A.wt <- enz[ which(enz$Protein == "E54AWT"), ]
enz.E54K.wt <- enz[ which(enz$Protein == "E54KWT"), ]
enz.E54Q.wt <- enz[ which(enz$Protein == "E54QWT"), ]
enz.Trnc.wt <- enz[ which(enz$Protein == "TrncWT"), ]
enz.E54A <- enz[ which(enz$Protein == "E54A"), ]
enz.E54K <- enz[ which(enz$Protein == "E54K"), ]
enz.E54Q <- enz[ which(enz$Protein == "E54Q"), ]
enz.trnc <- enz[ which(enz$Protein == "Trnc"), ]

# Generate enz sem and CI table for figure
enz.mean <- aggregate(enz$Absorbance ~ Protein, mean, data = enz)
enz.sem <- aggregate(enz$Absorbance ~ Protein, sem, data = enz)
LL.95 <- aggregate(enz$Absorbance ~ Protein, enz,
          FUN = function(x) t.test(x)$conf.int[1])
UL.95 <- aggregate(enz$Absorbance ~ Protein, enz,
          FUN = function(x) t.test(x)$conf.int[2])
enz.table <- data.frame(enz.mean[1], enz.mean[2], enz.sem[2],
          LL.95[2], UL.95[2])
colnames(enz.table) <- c("Protein", "mean", "sem", "LCI", "UCI")


# Generate Figure for structural diversity
#png(filename="~/../GitHub/Rpf/figures/Fig4.png",
#    width = 1200, height = 1200, res = 96*2)
#plot.new()
#par(oma=c(0,3,0,0), mar=c(2,2,0.5,6), mfrow = c(4,2))
#layout(matrix(c(1,5,2,6,3,7,4,8), 4, 2, byrow = TRUE))

# Alternative figure with no enz results #
png(filename="~/../GitHub/Rpf/figures/Fig4alt.png",
    width = 1200, height = 1200, res = 96*2)
plot.new()
par(oma=c(0,4,0,0), mar=c(3,4,2,1), mfrow=c(2,2))

# Plot final biomass growth curve results
plot(jitter(rep(2, length(od.E54A.wt$OD600)), amount = 0.1), od.E54A.wt$OD600, ylim = c(-0.2, 3.5), xlim = c(0.5, 5.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
box(lwd=2) 
points(jitter(rep(4, length(od.E54A$OD600)), amount = 0.1), od.E54A$OD600, pch = 21, bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

points(2, mean(od.E54A.wt$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)
points(4, mean(od.E54A$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.0", "1.0", "2.0", "3.0"), at = c(0, 1, 2, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(0, 1, 2, 3), labels = F)
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)

arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "E54AWT")], y1 = od.table$LCI[which(od.table$Treatment == "E54AWT")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "E54AWT")], y1 = od.table$UCI[which(od.table$Treatment == "E54AWT")], angle = 90,
       length=0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "E54A")], y1 = od.table$LCI[which(od.table$Treatment == "E54A")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "E54A")], y1 = od.table$UCI[which(od.table$Treatment == "E54A")], angle = 90,
       length=0.1, lwd = 2)

# Make x-axis labels
mtext('E54A', side = 1, line = 1.5, at = 4, cex = 1.5)
mtext('WT', side = 1, line = 1.5, at = 2, cex = 1.5)

plot(jitter(rep(2, length(od.E54K.wt$OD600)), amount = 0.1), od.E54K.wt$OD600, ylim = c(-0.2, 3.5), xlim = c(0.5, 5.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
box(lwd=2) 
points(jitter(rep(4, length(od.E54K$OD600)), amount = 0.1), od.E54K$OD600, pch = 21, bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

points(2, mean(od.E54K.wt$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)
points(4, mean(od.E54K$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.0", "1.0", "2.0", "3.0"), at = c(0, 1, 2, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(0, 1, 2, 3), labels = F)
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)

arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "E54KWT")], y1 = od.table$LCI[which(od.table$Treatment == "E54KWT")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "E54KWT")], y1 = od.table$UCI[which(od.table$Treatment == "E54KWT")], angle = 90,
       length=0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "E54K")], y1 = od.table$LCI[which(od.table$Treatment == "E54K")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "E54K")], y1 = od.table$UCI[which(od.table$Treatment == "E54K")], angle = 90,
       length=0.1, lwd = 2)

# Make x-axis labels
mtext('E54K', side = 1, line = 1.5, at = 4, cex = 1.5)
mtext('WT', side = 1, line = 1.5, at = 2, cex = 1.5)

plot(jitter(rep(2, length(od.E54Q.wt$OD600)), amount = 0.1), od.E54Q.wt$OD600, ylim = c(-0.2, 3.5), xlim = c(0.5, 5.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
box(lwd=2) 
points(jitter(rep(4, length(od.E54Q$OD600)), amount = 0.1), od.E54Q$OD600, pch = 21, bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

points(2, mean(od.E54Q.wt$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)
points(4, mean(od.E54Q$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.0", "1.0", "2.0", "3.0"), at = c(0, 1, 2, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(0, 1, 2, 3), labels = F)
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)

arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "E54QWT")], y1 = od.table$LCI[which(od.table$Treatment == "E54QWT")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "E54QWT")], y1 = od.table$UCI[which(od.table$Treatment == "E54QWT")], angle = 90,
       length=0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "E54Q")], y1 = od.table$LCI[which(od.table$Treatment == "E54Q")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "E54Q")], y1 = od.table$UCI[which(od.table$Treatment == "E54Q")], angle = 90,
       length=0.1, lwd = 2)

# Make x-axis labels
mtext('E54Q', side = 1, line = 1.5, at = 4, cex = 1.5)
mtext('WT', side = 1, line = 1.5, at = 2, cex = 1.5)

plot(jitter(rep(2, length(od.Trnc.wt$OD600)), amount = 0.1), od.Trnc.wt$OD600, ylim = c(-0.2, 3.5), xlim = c(0.5, 5.5), pch = 21, col = "lightgrey", bg = "lightgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
box(lwd=2) 
points(jitter(rep(4, length(od.trnc$OD600)), amount = 0.1), od.trnc$OD600, pch = 21, bg = "lightgrey", col = "lightgrey", lwd = 2, cex = 1.7)

points(4, mean(od.trnc$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)
points(2, mean(od.Trnc.wt$OD600), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.0", "1.0", "2.0", "3.0"), at = c(0, 1, 2, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(0, 1, 2, 3), labels = F)
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(2, 4), labels = F)

arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "Trnc")], y1 = od.table$LCI[which(od.table$Treatment == "Trnc")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(4), y0 = od.table$mean[which(od.table$Treatment == "Trnc")], y1 = od.table$UCI[which(od.table$Treatment == "Trnc")], angle = 90,
       length=0.1, lwd = 2)
arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "TrncWT")], y1 = od.table$LCI[which(od.table$Treatment == "TrncWT")], angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(2), y0 = od.table$mean[which(od.table$Treatment == "TrncWT")], y1 = od.table$UCI[which(od.table$Treatment == "TrncWT")], angle = 90,
       length=0.1, lwd = 2)

# Make x-axis labels
mtext('Trnc', side = 1, line = 1.5, at = 4, cex = 1.5)
mtext('WT', side = 1, line = 1.5, at = 2, cex = 1.5)
# Add outside Y axis 
mtext(expression('Biomass (OD600)'), side = 2,
      outer = TRUE, cex = 2, line = 1 , adj = 0.5)

# Plot EnzCheck Assay results
#plot(jitter(rep(2, length(enz.E54A$Absorbance)), amount = 0.1), enz.E54A$Absorbance, ylim = c(0, 100000), xlim = c(0.5, 5.5), pch = 21, col = "darkgrey", bg = "darkgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
#box(lwd=2) 
#points(jitter(rep(4, length(enz.E54A.wt$Absorbance)), amount = 0.1), enz.E54A.wt$Absorbance, pch = 21, bg = "darkgrey", col = "darkgrey", lwd = 2, cex = 1.7)

#points(2, mean(enz.E54A$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)
#points(4, mean(enz.E54A.wt$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)

#axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     labels = c("0", "25000", "50000", "75000"), at = c(0, 25000, 50000, 75000))
#axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at=c(0, 25000, 50000, 75000), labels = F)
#axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)
#axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)

#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "E54A")], y1 = enz.table$LCI[which(enz.table$Protein == "E54A")], angle = 90, length = 0.1, lwd = 2)
#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "E54A")], y1 = enz.table$UCI[which(enz.table$Protein == "E54A")], angle = 90,length=0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "E54AWT")], y1 = enz.table$LCI[which(enz.table$Protein == "E54AWT")], angle = 90, length = 0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "E54AWT")], y1 = enz.table$UCI[which(enz.table$Protein == "E54AWT")], angle = 90, length=0.1, lwd = 2)

# Make x-axis labels
#mtext('E54A', side = 1, line = 1, at = 2, cex = 1)
#mtext('WT', side = 1, line = 1, at = 4, cex = 1)

#plot(jitter(rep(2, length(enz.E54K$Absorbance)), amount = 0.1), enz.E54K$Absorbance, ylim = c(0, 100000), xlim = c(0.5, 5.5), pch = 21, col = "darkgrey", bg = "darkgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
#box(lwd=2) 
#points(jitter(rep(4, length(enz.E54K.wt$Absorbance)), amount = 0.1), enz.E54K.wt$Absorbance, pch = 21, bg = "darkgrey", col = "darkgrey", lwd = 2, cex = 1.7)

#points(2, mean(enz.E54K$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)
#points(4, mean(enz.E54K.wt$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)

#axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     labels = c("0", "25000", "50000", "75000"), at = c(0, 25000, 50000, 75000))
#axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at=c(0, 25000, 50000, 75000), labels = F)
#axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)
#axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)

#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "E54K")], y1 = enz.table$LCI[which(enz.table$Protein == "E54K")], angle = 90,
#       length = 0.1, lwd = 2)
#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "E54K")], y1 = enz.table$UCI[which(enz.table$Protein == "E54K")], angle = 90,
#       length=0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "E54KWT")], y1 = enz.table$LCI[which(enz.table$Protein == "E54KWT")], angle = 90,
#       length = 0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "E54KWT")], y1 = enz.table$UCI[which(enz.table$Protein == "E54KWT")], angle = 90,
#       length=0.1, lwd = 2)

# Make x-axis labels
#mtext('E54K', side = 1, line = 1, at = 2, cex = 1)
#mtext('WT', side = 1, line = 1, at = 4, cex = 1)

#plot(jitter(rep(2, length(enz.E54Q$Absorbance)), amount = 0.1), enz.E54Q$Absorbance, ylim = c(0, 100000), xlim = c(0.5, 5.5), pch = 21, col = "darkgrey", bg = "darkgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
#box(lwd=2) 
#points(jitter(rep(4, length(enz.E54Q.wt$Absorbance)), amount = 0.1), enz.E54Q.wt$Absorbance, pch = 21, bg = "darkgrey", col = "darkgrey", lwd = 2, cex = 1.7)

#points(2, mean(enz.E54Q$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)
#points(4, mean(enz.E54Q.wt$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)

#axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     labels = c("0", "25000", "50000", "75000"), at = c(0, 25000, 50000, 75000))
#axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at=c(0, 25000, 50000, 75000), labels = F)
#axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)
#axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)

#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "E54Q")], y1 = enz.table$LCI[which(enz.table$Protein == "E54Q")], angle = 90,
#       length = 0.1, lwd = 2)
#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "E54Q")], y1 = enz.table$UCI[which(enz.table$Protein == "E54Q")], angle = 90,
#       length=0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "E54QWT")], y1 = enz.table$LCI[which(enz.table$Protein == "E54QWT")], angle = 90,
#       length = 0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "E54QWT")], y1 = enz.table$UCI[which(enz.table$Protein == "E54QWT")], angle = 90,
#       length=0.1, lwd = 2)

# Make x-axis labels
#mtext('E54Q', side = 1, line = 1, at = 2, cex = 1)
#mtext('WT', side = 1, line = 1, at = 4, cex = 1)

#plot(jitter(rep(2, length(enz.trnc$Absorbance)), amount = 0.1), enz.trnc$Absorbance, ylim = c(0, 100000), xlim = c(0.5, 5.5), pch = 21, col = "darkgrey", bg = "darkgrey", lwd = 2, cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis= 1.5, las = 1, ylab = "", xlab = "")
#box(lwd=2) 
#points(jitter(rep(4, length(enz.Trnc.wt$Absorbance)), amount = 0.1), enz.Trnc.wt$Absorbance, pch = 21, bg = "darkgrey", col = "darkgrey", lwd = 2, cex = 1.7)

#points(2, mean(enz.trnc$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)
#points(4, mean(enz.Trnc.wt$Absorbance), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)

#axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     labels = c("0", "25000", "50000", "75000"), at = c(0, 25000, 50000, 75000))
#axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at=c(0, 25000, 50000, 75000), labels = F)
#axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)
#axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
#     at = c(2, 4), labels = F)

#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "Trnc")], y1 = enz.table$LCI[which(enz.table$Protein == "Trnc")], angle = 90,
#       length = 0.1, lwd = 2)
#arrows(x0 = c(2), y0 = enz.table$mean[which(enz.table$Protein == "Trnc")], y1 = enz.table$UCI[which(enz.table$Protein == "Trnc")], angle = 90,
#       length=0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "WT")], y1 = enz.table$LCI[which(enz.table$Protein == "WT")], angle = 90,
#       length = 0.1, lwd = 2)
#arrows(x0 = c(4), y0 = enz.table$mean[which(enz.table$Protein == "WT")], y1 = enz.table$UCI[which(enz.table$Protein == "WT")], angle = 90,
#       length=0.1, lwd = 2)

# Make x-axis labels
#mtext('Trnc', side = 1, line = 1, at = 2, cex = 1)
#mtext('WT', side = 1, line = 1, at = 4, cex = 1)

# Add outside Y axis 
#mtext(expression('Relative Final Fluorescence (Absorbance)'), side = 2,
#      outer = TRUE, cex = 1.5, line = -20 , adj = 0.5)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
#img <- readPNG("figures/Fig4.png")
#grid.raster(img)
img <- readPNG("figures/Fig4alt.png")
grid.raster(img)
```


## 6) Effects of recombinant Rpf from KBS0714 on diverse bacterial strains

```{r}
# A) Generate % change in growth curve parameter with Rpf treatment among KBS0714 strains # 
# Load gompertz output data 
gc.param.1 <- read.delim("data/generality/20160320_generality_gomp_param_output.txt", sep = ",", header = TRUE)
gc.param.2 <- read.delim("data/generality/20160323_generality_gomp_param_output.txt", sep = ",", header = TRUE)
# Load well key for growth curve assay 
gc.key <- read.delim("data/generality/generality.gc.well.key.txt", sep = "\t", header = TRUE)

# Label phylogenetic and treatment information in datasets 
key.1 <- subset(gc.key, Run_Num == "1", select = c(Strain,Gram,Phylum,Genus,Treatment,Rep))
gc.param.1 <- cbind(gc.param.1,key.1)
key.2 <- subset(gc.key, Run_Num == "2", select = c(Strain,Gram,Phylum,Genus,Treatment,Rep))
gc.param.2 <- cbind(gc.param.2,key.2)

# Merge datasets
gc <- rbind(gc.param.1,gc.param.2)
# Remove unused parameters to simplify
gc <- gc[ ,c(-(1:2),-(6:10))]
# Some biomass yeild values are negative, force to be positive # 
gc$A <- abs(gc$A)

# Calculate % Biomass yeild (A) change
a <- as.data.frame(gc$A[which(gc$Treatment == "Rpf+")])
b <- as.data.frame(gc$A[which(gc$Treatment == "Rpf-")])
df <- c()
df <- as.data.frame(((a-b)/b)*100)
# Calculate % Maximum growth rate (umax) change
c <- as.data.frame(gc$umax[which(gc$Treatment == "Rpf+")])
d <- as.data.frame(gc$umax[which(gc$Treatment == "Rpf-")])
df.u <- as.data.frame(((c-d)/d)*100)
df <- cbind(df, df.u)
# Calculate % Lag time (L) change
e <- gc$L[which(gc$Treatment == "Rpf+")]
f <- gc$L[which(gc$Treatment == "Rpf-")]
df.l <- as.data.frame(((e-f)/f)*100)
df <- cbind(df, df.l)
colnames(df) <- c("A", "umax", "L")

# Add phylogenetic information to dataset
df.info.1 <- as.data.frame(gc[1:24, 4:7])
df.info.2 <- as.data.frame(gc[73:96, 4:7])
df.info <- rbind(df.info.1, df.info.2)
df <- cbind(df, df.info) 

# Remove outlier data points: Reps with some parameter exceeding 100% change
#df <- df[c(-5, -6, -7, -8, -32, -33, -34), ] # Remove Art1
#df <- df[c(-6, -8), ]
df <- df[c(-5, -6, -7, -8), ] # Only remove art1

df.norm <- as.data.frame(scale(df[,1:3]))
df.norm <- cbind(df.norm, df[, 4:7])


# Generate L table of means, sems, and CIs 
df.means <- aggregate(L ~ Genus, mean, data= df.norm)
df.sems <- aggregate(L ~ Genus, sem, data = df.norm)
df.LL.95 <- aggregate(df.norm$L ~ Genus, df.norm,
          FUN = function(x) t.test(x)$conf.int[1])
df.UL.95 <- aggregate(df.norm$L ~ Genus, df.norm,
          FUN = function(x) t.test(x)$conf.int[2])
df.table <- data.frame(df.means[1], df.means[2], df.sems[2],
                       df.LL.95[2], df.UL.95[2])
colnames(df.table) <- c( "Genus",  "mean", "sem", "LCI", "UCI")

# Order df.table
label <- c(2,7,6,3,8,1,5,11,9,4,10)
df.table$PD <- label
df.table <- df.table[order(df.table$PD), ]


# Generate A table of means, sems, and CIs
df.A.means <- aggregate(A ~ Genus, mean, data= df.norm)
df.A.sems <- aggregate(A ~ Genus, sem, data = df.norm)
df.A.LL.95 <- aggregate(df.norm$A ~ Genus, df.norm,
          FUN = function(x) t.test(x)$conf.int[1])
df.A.UL.95 <- aggregate(df.norm$A ~ Genus, df.norm,
          FUN = function(x) t.test(x)$conf.int[2])
df.A.table <- data.frame(df.A.means[1], df.A.means[2], df.A.sems[2],
                       df.A.LL.95[2], df.A.UL.95[2])
colnames(df.A.table) <- c( "Genus",  "mean", "sem", "LCI", "UCI")

# Order df.table
label <- c(2,7,6,3,8,1,5,11,9,4,10)
df.A.table$PD <- label
df.A.table <- df.A.table[order(df.A.table$PD), ]


# Generate umax table of means, sems, and CIs
df.umax.means <- aggregate(umax ~ Genus, mean, data= df.norm)
df.umax.sems <- aggregate(umax ~ Genus, sem, data = df.norm)
df.umax.LL.95 <- aggregate(df.norm$umax ~ Genus, df.norm,
          FUN = function(x) t.test(x)$conf.int[1])
df.umax.UL.95 <- aggregate(df.norm$umax ~ Genus, df.norm,
          FUN = function(x) t.test(x)$conf.int[2])
df.umax.table <- data.frame(df.umax.means[1], df.umax.means[2], df.umax.sems[2],
                       df.umax.LL.95[2], df.umax.UL.95[2])
colnames(df.umax.table) <- c( "Genus",  "mean", "sem", "LCI", "UCI")

# Order df.table
label <- c(2,7,6,3,8,1,5,11,9,4,10)
df.umax.table$PD <- label
df.umax.table <- df.umax.table[order(df.umax.table$PD), ]



# Subset data for figure 
df.norm.Ped <- df.norm[ which(df.norm$Genus == "Pedobacter"), ]
#df.norm.Art1 <- df.norm[ which(df.norm$Genus == "Arthrobacter1"), ]
df.norm.Art2 <- df.norm[ which(df.norm$Genus == "Arthrobacter2"), ]
df.norm.Myc <- df.norm[ which(df.norm$Genus == "Mycobacterium"), ]
df.norm.Azo <- df.norm[ which(df.norm$Genus == "Azospirillium"), ]
df.norm.Rho <- df.norm[ which(df.norm$Genus == "Rhodococcus"), ]
df.norm.Pse <- df.norm[ which(df.norm$Genus == "Pseudomonas"), ]
df.norm.Jan <- df.norm[ which(df.norm$Genus == "Janthinobacterium"), ]
df.norm.Var <- df.norm[ which(df.norm$Genus == "Variovorax"), ]
df.norm.Mic <- df.norm[ which(df.norm$Genus == "Micrococcus"), ]
df.norm.Cur <- df.norm[ which(df.norm$Genus == "Curtobacterium"), ]
df.norm.Bac <- df.norm[ which(df.norm$Genus == "Bacillus"), ]


# Generate figure 
#png(filename="~/../GitHub/Rpf/figures/Fig6.png",  
#    width = 1200, height = 1200, res = 96*2)
#plot.new()
#par(oma=c(3,5,0,0), mar=c(1,2,1,1), mfrow = c(2,1),  byrow = TRUE)
```


```{r}


# Generate Alt. 3x1 figure
png(filename="~/../GitHub/Rpf/figures/Fig6Alt.png",  
    width = 1200, height = 1200, res = 96*2)
plot.new()
par(oma=c(3,5,0,0), mar=c(1,2,1,1), mfrow = c(3,1),  byrow = TRUE)

# Lag time
plot(jitter(rep(1, length(df.norm.Mic$L)), amount = 0.1), df.norm.Mic$L,
     ylim = c(-3, 3.5), xlim = c(0.5, 11.5), pch = 21, col = "grey", bg = "grey", lwd = 2,
           cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 1.5,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
#points(jitter(rep(2, length(df.norm.Art1$L)), amount = 0.1), df.norm.Art1$L, pch = 21, 
#       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(2, length(df.norm.Art2$L)), amount = 0.1), df.norm.Art2$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(3, length(df.norm.Cur$L)), amount = 0.1), df.norm.Cur$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(4, length(df.norm.Rho$L)), amount = 0.1), df.norm.Rho$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(5, length(df.norm.Myc$L)), amount = 0.1), df.norm.Myc$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(6, length(df.norm.Bac$L)), amount = 0.1), df.norm.Bac$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(7, length(df.norm.Azo$L)), amount = 0.1), df.norm.Azo$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(8, length(df.norm.Jan$L)), amount = 0.1), df.norm.Jan$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(9, length(df.norm.Pse$L)), amount = 0.1), df.norm.Pse$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(10, length(df.norm.Var$L)), amount = 0.1), df.norm.Var$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(11, length(df.norm.Ped$L)), amount = 0.1), df.norm.Ped$L, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)

points(1, mean(df.norm.Mic$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
#points(2, mean(df.norm.Art1$L), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)  
points(2, mean(df.norm.Art2$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(3, mean(df.norm.Cur$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(4, mean(df.norm.Rho$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(5, mean(df.norm.Myc$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  
points(6, mean(df.norm.Bac$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(7, mean(df.norm.Azo$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(8, mean(df.norm.Jan$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(9, mean(df.norm.Pse$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(10, mean(df.norm.Var$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(1, mean(df.norm.Ped$L), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 

# Y-axis label
mtext(expression('Lag Time (L)'), side = 2,
      outer = FALSE, cex = 1.25, line = 3, adj = 0.5)

# Major Axes
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = F, at = c(1, 2, 3,4,5,6,7,8,9,10,11), las =3)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-3.0", "-2.0", "-1.0", "0.0", "1.0", "2.0", "3.0"), at = c(-3, -2, -1, 0, 1, 2, 3))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(-3, -2, -1, 0, 1, 2, 3), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(1, 2, 3,4,5,6,7,8,9,10,11), labels = F)

# Add dotted line at zero
abline(h=0, col="black", lty =2, lwd=2)

arrows(x0 = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), y0 = df.table$mean, y1 = df.table$LCI, angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), y0 = df.table$mean, y1 = df.table$UCI, angle = 90,
       length=0.1, lwd = 2)

box(lwd = 2)


# Biomass Yield
plot(jitter(rep(1, length(df.norm.Mic$A)), amount = 0.1), df.norm.Mic$A,
     ylim = c(-3, 3.5), xlim = c(0.5, 11.5), pch = 21, col = "grey", bg = "grey", lwd = 2,
           cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 1.5,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
#points(jitter(rep(2, length(df.norm.Art1$A)), amount = 0.1), df.norm.Art1$A, pch = 21, 
#       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(2, length(df.norm.Art2$A)), amount = 0.1), df.norm.Art2$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(3, length(df.norm.Cur$A)), amount = 0.1), df.norm.Cur$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(4, length(df.norm.Rho$A)), amount = 0.1), df.norm.Rho$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(5, length(df.norm.Myc$A)), amount = 0.1), df.norm.Myc$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(6, length(df.norm.Bac$A)), amount = 0.1), df.norm.Bac$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(7, length(df.norm.Azo$A)), amount = 0.1), df.norm.Azo$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(8, length(df.norm.Jan$A)), amount = 0.1), df.norm.Jan$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(9, length(df.norm.Pse$A)), amount = 0.1), df.norm.Pse$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(10, length(df.norm.Var$A)), amount = 0.1), df.norm.Var$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(11, length(df.norm.Ped$A)), amount = 0.1), df.norm.Ped$A, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)

points(1, mean(df.norm.Mic$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
#points(2, mean(df.norm.Art1$A), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)  
points(2, mean(df.norm.Art2$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(3, mean(df.norm.Cur$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(4, mean(df.norm.Rho$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(5, mean(df.norm.Myc$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  
points(6, mean(df.norm.Bac$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(7, mean(df.norm.Azo$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(8, mean(df.norm.Jan$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(9, mean(df.norm.Pse$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(10, mean(df.norm.Var$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(11, mean(df.norm.Ped$A), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 

# Outer Y-axis label
mtext(expression('Normalized % Change with Rpf Treatment'), side = 2,
      outer = TRUE, cex = 1.5, line = 3, adj = 0.5)
# Y-axis label
mtext(expression('Biomass Yield (A)'), side = 2,
      outer = FALSE, cex = 1.25, line = 3, adj = 0.5)

# Major Axes
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels =  FALSE, at = c(1, 2, 3,4,5,6,7,8,9,10,11), las =3)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-3.0", "-2.0", "-1.0", "0.0", "1.0", "2.0", "3.0"), at = c(-3, -2, -1, 0, 1, 2, 3))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(-3, -2, -1, 0, 1, 2, 3), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(1, 2, 3,4,5,6,7,8,9,10,11), labels = F)

# Add dotted line at zero
abline(h=0, col="black", lty =2, lwd=2)

arrows(x0 = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), y0 = df.A.table$mean, y1 = df.A.table$LCI, angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), y0 = df.A.table$mean, y1 = df.A.table$UCI, angle = 90,
       length=0.1, lwd = 2)

box(lwd = 2)


# Maximum growth rate (umax)
plot(jitter(rep(1, length(df.norm.Mic$umax)), amount = 0.1), df.norm.Mic$umax,
     ylim = c(-3, 3.5), xlim = c(0.5, 11.5), pch = 21, col = "grey", bg = "grey", lwd = 2,
           cex = 1.7, yaxt = "n", xaxt = "n", cex.lab = 2, cex.axis = 1.5,
      las = 1, ylab = "", xlab = "")
      box(lwd = 2)
#points(jitter(rep(2, length(df.norm.Art1$A)), amount = 0.1), df.norm.Art1$A, pch = 21, 
#       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(2, length(df.norm.Art2$umax)), amount = 0.1), df.norm.Art2$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(3, length(df.norm.Cur$umax)), amount = 0.1), df.norm.Cur$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(4, length(df.norm.Rho$umax)), amount = 0.1), df.norm.Rho$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(5, length(df.norm.Myc$umax)), amount = 0.1), df.norm.Myc$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(6, length(df.norm.Bac$umax)), amount = 0.1), df.norm.Bac$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(7, length(df.norm.Azo$umax)), amount = 0.1), df.norm.Azo$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(8, length(df.norm.Jan$umax)), amount = 0.1), df.norm.Jan$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(9, length(df.norm.Pse$umax)), amount = 0.1), df.norm.Pse$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(10, length(df.norm.Var$umax)), amount = 0.1), df.norm.Var$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)
points(jitter(rep(11, length(df.norm.Ped$umax)), amount = 0.1), df.norm.Ped$umax, pch = 21, 
       bg = "grey", col = "grey", lwd = 2, cex = 1.7)

points(1, mean(df.norm.Mic$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
#points(2, mean(df.norm.Art1$umax), pch = 21, col = "black", 
#       bg = "NA", lwd = 2, cex = 2.5)  
points(2, mean(df.norm.Art2$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(3, mean(df.norm.Cur$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(4, mean(df.norm.Rho$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(5, mean(df.norm.Myc$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5)  
points(6, mean(df.norm.Bac$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(7, mean(df.norm.Azo$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(8, mean(df.norm.Jan$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(9, mean(df.norm.Pse$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(10, mean(df.norm.Var$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 
points(11, mean(df.norm.Ped$umax), pch = 21, col = "black", 
       bg = "NA", lwd = 2, cex = 2.5) 

# Y-axis label
mtext(expression('Umax'), side = 2,
      outer = FALSE, cex = 1.25, line = 3, adj = 0.5)

# Major Axes
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("Mic", "Art2", "Cur", "Rho", "Myc", "Bac", "Azo", "Jan", "Pse", "Var", "Ped"), 
     at = c(1, 2, 3,4,5,6,7,8,9,10,11), las =3)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-3.0", "-2.0", "-1.0", "0.0", "1.0", "2.0", "3.0"), at = c(-3, -2, -1, 0, 1, 2, 3))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(-3, -2, -1, 0, 1, 2, 3), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(1, 2, 3,4,5,6,7,8,9,10,11), labels = F)

# Add dotted line at zero
abline(h=0, col="black", lty =2, lwd=2)

arrows(x0 = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), y0 = df.umax.table$mean, y1 = df.umax.table$LCI, angle = 90,
       length = 0.1, lwd = 2)
arrows(x0 = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11), y0 = df.umax.table$mean, y1 = df.umax.table$UCI, angle = 90,
       length=0.1, lwd = 2)

box(lwd = 2)


# Close Plot Device
dev.off()
graphics.off()

# Show Plot
#img <- readPNG("~/../GitHub/Rpf/figures/Fig6.png")
#grid.raster(img)

img <- readPNG("~/../GitHub/Rpf/figures/Fig6Alt.png")
grid.raster(img)

```




```{r}
# A) Generate simple bargraph to look at the effects of Rpf on growth curve parameters
library(ggplot2)

# Calculate mean and sem 
# Biomass yeild (A)
generality.A.means.sem <- ddply(gc, c("Genus","Treatment"), summarise,
                   mean=mean(A), sem=sd(A)/sqrt(length(A)))
generality.A.means.sem <- transform(generality.A.means.sem, lower=mean-sem, upper=mean+sem)
# Maximum growth rate (umax)
generality.umax.means.sem <- ddply(gc, c("Genus", "Treatment"), summarise,
                   mean=mean(umax), sem=sd(umax)/sqrt(length(umax)))
generality.umax.means.sem <- transform(generality.umax.means.sem, lower=mean-sem, upper=mean+sem)
# Lag time (L)
generality.L.means.sem <- ddply(gc, c("Genus","Treatment"), summarise,
                   mean=mean(L), sem=sd(L)/sqrt(length(L)))
generality.L.means.sem <- transform(generality.L.means.sem, lower=mean-sem, upper=mean+sem)


# Bargraph figure 
A.bar <- ggplot(data=generality.A.means.sem, aes(x=Genus, y=mean, fill=Treatment)) + 
  geom_bar(colour="black", position = position_dodge(), stat = "identity", size = 1)

A.bar + geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(1), 
                          data=generality.A.means.sem, width = 0.35, size=1) +
  labs(x =  "Genus" ,y="Biomass Yield") +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size = 12),
        axis.text.x=element_text(colour="black", size=12, angle =45,hjust = 1),
        axis.title = element_text(size=18),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2))

umax.bar <- ggplot(data=generality.umax.means.sem, aes(x=Genus, y=mean, fill=Treatment)) + 
  geom_bar(colour="black", position = position_dodge(), stat = "identity", size = 1)

umax.bar + geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(1), 
                          data=generality.umax.means.sem, width = 0.35, size=1) +
  labs(x =  "Genus" ,y="Maximum Growth Rate") +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size = 12),
        axis.text.x=element_text(colour="black", size=12, angle =45,hjust = 1),
        axis.title = element_text(size=18),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2))

L.bar <- ggplot(data=generality.L.means.sem, aes(x=Genus, y=mean, fill=Treatment)) + 
  geom_bar(colour="black", position = position_dodge(), stat = "identity", size = 1)

L.bar + geom_errorbar(aes(ymax=upper, ymin=lower), position=position_dodge(1), 
                          data=generality.L.means.sem, width = 0.35, size=1) +
  labs(x =  "Genus" ,y="Lag Time") +
  theme_classic() +
  theme(axis.text.y=element_text(colour="black", size = 12),
        axis.text.x=element_text(colour="black", size=12, angle =45,hjust = 1),
        axis.title = element_text(size=18),
        axis.title.y = element_text(colour="black",  margin = margin(0,10,0,10)),
        axis.title.x = element_text(colour="black",margin=margin(15,10,0,10)),
        panel.border = element_rect(linetype = "solid", colour = "black", size=3, fill=NA),
        axis.ticks.length = unit(.25, "cm"),
        axis.ticks = element_line(size = 2))

```


# 7) Effect of phylogeny on response to Rpf 

```{r}
# A) Phylognetic analysis of generality assay results 
# Load required packages 
require('ape')
require('seqinr')
require('phytools')
require('phangorn')
require('phylobase')
require('adephylo')
require('geiger')
require('picante')
require('stats')
require('caper')
require('reshape')
require('muscle')
require('RColorBrewer')

# Construct traits dataset used for phylogenetic trait analysis 
# Averaging traits by genus 
df.A.means <- ddply(df, c("Genus"), summarise,
                   mean=mean(A), sem=sd(A)/sqrt(length(A)))
df.umax.means <- ddply(df, c("Genus"), summarise,
                   mean=mean(umax), sem=sd(umax)/sqrt(length(umax)))
df.L.means <- ddply(df, c("Genus"), summarise,
                   mean=mean(L), sem=sd(L)/sqrt(length(L)))

# Combine datasets to construct traits dataset
traits <- cbind(df.A.means,df.umax.means,df.L.means)

# Construct sem dataset 
sem <- traits[ ,c(1,3,6,9)]
colnames(sem) <- c("Genus", "A", "umax", "L")

# Construct traits dataset
traits <- traits[ , c(1,2,5,8)]   # The trait values do not make sense. Contradiction in KBS0714 results
colnames(traits) <- c("Genus", "A", "umax", "L")

# Make Genus names as row names 
row.names(traits) <- traits$Genus
# Remove Genus name column
traits <- traits[, -1]

# Standardize traits (mean=0, std = 1)
traits.norm <- scale(traits)

# Check scaling
check.norm.mean <- round(colMeans(traits.norm, na.rm = TRUE), 1)
check.norm.sd <- apply(traits.norm, 2, sd, na.rm = TRUE)

# Convert scaled list back to dataframe
traits.norm.db <- data.frame(matrix(unlist(traits.norm), 
              nrow=11, byrow = T), stringsAsFactors = FALSE)
rownames(traits.norm.db) <- rownames(traits)
colnames(traits.norm.db) <- colnames(traits.norm)

# Add Genus names back to traits.norm
traits.norm <- as.data.frame(traits.norm)
traits.norm$Genus <- traits$Genus
#traits.norm <- traits.norm[c(4,1,2,3)]

###########
# Read FASTA file as DNAStringSet 
dna <- readDNAStringSet("KBSsequences.fasta")

# Performing an Alignment using Muscle from Ape package 
aln <- muscle::muscle(seqs="dna.fasta", stringset = dna, out = "dna.afa")

# Convert Alignment File to DNAbin Object 
dna.bin <- as.DNAbin(aln)
dna.bin.1 <- as.phyDat(aln)

# Create Distance Matrix with "raw" Model {ape}
seq.dist.raw <- dist.dna(dna.bin, model = "raw", pairwise.deletion = FALSE)

# Neighbor Joining Algorithm to Construct Tree, a 'phylo' Object {ape}
nj.tree <- bionj(seq.dist.raw)

# Identify Outgroup Sequence
outgroup <- match("Methanosarcina", nj.tree$tip.label)

# Root the Tree {ape}
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)

# Plot the Rooted Tree{ape}
par(mar = c(1,1,2,1) + 0.1)
plot.phylo(nj.rooted, main = "Neigbor Joining Tree", "phylogram", use.edge.length = FALSE,
           direction = "right", cex = 1, label.offset = 1)
add.scale.bar(cex = 0.7)

# Generate Neighbor Joining Tree Using F84 DNA Substitution Model {ape}
nj.tree <- bionj(seq.dist.F84)

# Define the Outgroup
outgroup <- match("Methanosarcina", nj.tree$tip.label)

# Create a Rooted Tree {ape}
nj.rooted <- root(nj.tree, outgroup, resolve.root = TRUE)

# Keep Rooted but Drop Outgroup Branch
nj.rooted <- drop.tip(nj.rooted, "Methanosarcina")

# Define Color Palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"))

# Normalized traits with na.omit
traits.norm.na <- na.omit(traits.norm)

# Map Growth curve parameter response to Rpf treatment {adephylo}
par(mar=c(1,1,1,1) + 0.1)
x <- phylo4d(nj.rooted, traits.norm.na)
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE,
              cex.label = 0.8, scale = TRUE, use.edge.length = FALSE,
              edge.color = "black", edge.width = 2, box = FALSE,
              col=mypalette(25), pch = 15, cex.symbol = 2.5,
              ratio.tree = 0.75, cex.legend = 1, center = FALSE)
##################################################


# Below part should not work becuase traits should be missing Art1 now

# Generating Phylogenetic ML tree
# Load ML tree 
ml.tree <- read.tree("data/generality/phylo/RAxML_bipartitionsBranchLabels.KBS.ML")

# Define the outgroup
outgroup <- match("Methanosarcina", ml.tree$tip.label)

# Create a rooted tree {ape}
ml.rooted <- root(ml.tree, outgroup, resolve.root = TRUE)

# Keep rooted but drop outgroup branch
ml.rooted <- drop.tip(ml.rooted, "Methanosarcina")

# Define Color Palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"))

# Map traits {adephylo}
par(mar=c(1,1,1,1) + 0.1)
x <- phylo4d(ml.rooted, traits.norm) 
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE, 
              cex.label = 0.8, scale = TRUE, use.edge.length = FALSE,
              edge.color = "black", edge.width = 3, box = FALSE,
              col=mypalette(25), pch = 15, cex.symbol = 2.5,
              ratio.tree = 0.7, cex.legend = 1, center = TRUE)


# C) Does rpf effect on growth parameters vary among species (ANOVA)
# two-factor ANOVA
generality.umax.anova <- aov(umax ~ Treatment*Genus, data = gc)
summary(generality.umax.anova)
    # Non-significant main effect of Rpf treatment on umax (p=0.19)
    # Highly significant effect of genus (p < 0.001)
    # Significant effect of Treatment and strain interaction (p=0.00406)
generality.L.anova <- aov(L ~ Treatment*Genus, data = gc)
summary(generality.L.anova)
    # Significant effect of Rpf treatment on L (p=0.0085)
    # Highly significant effect of genus (p < 0.001)
    # Non-significant effect of treatment and strain interaction (p=0.318)
generality.A.anova <- aov(A ~ Treatment*Genus, data = gc)
summary(generality.A.anova)
    # Highly Significant effect of Rpf treatment (p < 0.001)
    # Highly significant effect of strain (p < 0.001)
    # Highly significant interaction effect of Rpf and strain (p < 0.001)
      # Reject null hypothesis --> Rpf differentially affected strains growth parameters

# D) Statistical tests: Are effects influenced by whether strain is Gram + vs. Gram -?
# two-factor ANOVA
# Gram status 
generality.g.umax.anova <- aov(umax ~ Treatment*Gram, data = gc)
summary(generality.g.umax.anova)
    # Non-significant main effect of Rpf (p=0.4676)
    # Highly significant main effect of Gram (p<0.001)
    # Significant interaction effect of Rpf and gram (p=0.0486)
generality.g.L.anova <- aov(L ~ Treatment*Gram, data = gc)
summary(generality.g.L.anova)
    # Non-significant main effect of Rpf treatment (p=0.52)
    # Significant main effect of Gram status on Lag time (p=0.0007)
    # Non-significant interaction effect (0.64)
generality.g.A.anova <- aov(A ~ Treatment*Gram, data = gc)
summary(generality.g.A.anova)
    # Non-significant main effect of treatment 
    # Non-significant main effect of Gram and interaction
      # Fail to reject null hypothesis --> Gram status does not influence response to Rpf 


# E) Statistical test: Testing for Phylogenetic signal
# Pagel's Lambda 
# Rescale tree
ml.lambda.5 <- rescale(ml.rooted, "lambda", 0.5)
ml.lambda.0 <- rescale(ml.rooted, "lambda", 0)

# Pagel lambda for growth curve response
lambda.a.model <- fitContinuous(ml.rooted, traits.norm, model = "lambda")
lambda.a.model.5 <- fitContinuous(ml.lambda.5, traits.norm, model = "lambda")
lambda.a.model.0 <- fitContinuous(ml.lambda.0, traits.norm, model = "lambda")
lambda.a.model      
lambda.a.model.5
lambda.a.model.0 
    # The outputs are al the same --> No phylogenetic signal of Rpf effects on species growth


# Blomberg's K 
ml.rooted$edge.length <- ml.rooted$edge.length + 10^-7
traits.p <- matrix(NA, 6, 3)
colnames(traits.p) <- colnames(traits.norm)
rownames(traits.p) <- c("K", "PIC.var.obs", "PIC.var.mean", "PIC.var.P", "PIC.var.z", "PIC.P.BH")

for (i in 1:3) {
  x <- as.matrix(traits.norm[ ,i, drop = FALSE])
  out <- phylosignal(x, ml.rooted)
  traits.p[1:5, i] <- round(t(out), 3)
}

traits.p[6, ] <- round(p.adjust(traits.p[4, ], method = "BH"), 3)
traits.p # Look at the output of Blomberg's K
    # PIC.P.BH corrected p value show non-significant effects of Rpf on growth parameters


#F) Generate figure to show Rpf effects on phylogeny 
#sem <- read.table("sem.txt", sep = "\t", header = TRUE)

#traits.L <- traits[ , 3]
L <- as.data.frame(traits.norm$L)
row.names(L)<- row.names(traits)
name <- as.character("L")
colnames(L) <- name

#traits.A <- traits[ , 1]
A <- as.data.frame(traits.norm$A)
row.names(A)<- row.names(traits)
name <- as.character("A")
colnames(A) <- name

#traits.u <- traits[ , 2]
u <- as.data.frame(traits.norm$umax)
row.names(u)<- row.names(traits)
name <- as.character("umax")
colnames(u) <- name
  
# Plot barplot of  proportional changes 
# Lag time
tree.L <- plotTree.barplot(ml.rooted, L , args.barplot = list(beside=TRUE, xlim = c(-2,3), 
    legend.text=TRUE,space=c(0,0.5),args.legend=list(x=1,y=40),
    xlab="% Change in Lag time with Rpf"))
# Biomass yield
tree.A <- plotTree.barplot(ml.rooted, A , args.barplot = list(beside=TRUE, xlim = c(-2,3), 
    legend.text=TRUE,space=c(0,0.5),args.legend=list(x=1,y=40),
    xlab="% Change in biomass yield with Rpf"))
# Maximum growth rate
tree.u <- plotTree.barplot(ml.rooted, u , args.barplot = list(beside=TRUE, xlim = c(-1.5,2), 
    legend.text=TRUE,space=c(0,0.5),args.legend=list(x=1,y=40),
    xlab="% Change in maximum growth rate with Rpf"))
```

















