---
title: "Phylogenetic analysis of resuscitation"
author: "Jay T. Lennon"
date: "June 4, 2020"
header-includes:
  - \usepackage{array}
output: pdf_document
geometry: margin=2.54cm
---

## 1) SETUP

### A. Retrieve and Set Your Working Directory

```{r, results = 'hide'}
rm(list = ls())
getwd()
setwd("~/GitHub/Rpf/")

### B. Load Packages 
Several R packages have been developed for conducting phylogenetic analyses (http://goo.gl/DtU16j).
We will describe the packages in greater details in the sections below.
The `require()` function in `R` returns `TRUE` if the package was successfully loaded or `FALSE` if the package failed to load. 
This `for` loop loads each package and installs the package when `require()` returns `FALSE`.

```{r, results = 'hide', message = FALSE, warning = FALSE, tidy=TRUE, tidy.opts=list(width.cutoff=60)} 
package.list <- c('ape', 'seqinr', 'phylobase', 'adephylo', 'geiger', 
                  'picante', 'stats', 'RColorBrewer', 'caper', 'phylolm', 'pmc', 
                  'ggplot2', 'tidyr', 'dplyr', 'phangorn', 'pander', 'muscle') 
for (package in package.list) {
  if (!require(package, character.only=TRUE, quietly=TRUE)) {
    install.packages(package)
    library(package, character.only=TRUE)
  }
}

### C. Align sequences 
##[SO FAR ONLY APE AND SEQUINR REQUIRED]

# Ran multisequence alignment on local machine using muscle

```{bash, results = 'hide', eval = FALSE}
muscle -in ./data/phylogeny/Rpf.16S.tree.fasta-out ./data/phylogeny/Rpf.16S.tree.afa
```

# View alignment

```{r}
# Read Alignment File {seqinr}
read.aln <- read.alignment(file = "./data/phylogeny/Rpf.16S.tree.afa", format = "fasta")  

# Convert Alignment File to DNAbin Object {ape}
rpf.DNAbin <- as.DNAbin(read.aln) 

# Identify Base Pair Region of 16S rRNA Gene to Visuzlize
window <- rpf.DNAbin[, 100:500] 

# Command to Visusalize Sequence Alignment {ape}
image.DNAbin(window, cex.lab = 0.50) 
```


### D. Create ML tree with bootstrap values

```{r}
# Read Alignment File {seqinr}

ml.new <- read.tree("./data/phylogeny/RAxML_bipartitions.rpf.ml")

par(mar = c(1,1,2,1) + 0.1)
plot.phylo(ml.new, type = "phylogram", direction = "right", show.tip.label=TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1, 
           main = "Maximum Likelihood with Support Values")
add.scale.bar(cex = 0.7)
nodelabels(ml.new$node.label, font = 2, bg = "white", frame = "r", cex = 0.5)
```

### E. Map traits (i.e., growth-parameter responses to Rpf) onto tree

ml.new.rooted <- root(ml.new, "Methanosarcina")
ml.map <- ape::drop.tip(ml.new.rooted, "Methanosarcina")

par(mar = c(1,2,2,1) + 0.1)
plot.phylo(ml.bootstrap, type = "phylogram", direction = "right", show.tip.label=TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1, 
           main = "Maximum Likelihood with Support Values")
add.scale.bar(cex = 0.7)
nodelabels(ml.bootstrap$node.label, font = 2, bg = "white", frame = "r", cex = 0.5)

mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"))

# Map Phosphorus Traits {adephylo}
par(mar=c(1,1,1,1) + 0.1)
x <- phylo4d(nj.rooted, p.growth.std)
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = TRUE, 
              cex.label = 0.5, scale = FALSE, use.edge.length = FALSE, 
              edge.color = "black", edge.width = 2, box = FALSE, 
              col=mypalette(25), pch = 15, cex.symbol = 1.25, 
              ratio.tree = 0.5, cex.legend = 1.5, center = FALSE)   


