---
title: "Rpf: generality"
author: "Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
output: pdf_document
geometry: margin=2.54cm
---

Termination of dormancy in soil bacteria with resuscitation promoting factor (Rpf)

## Clear workspace and set directory

```{r setup}
rm(list=ls())
getwd()
setwd("~/GitHub/Rpf")
```

## Load packages and functions

```{r}
library("png")
library("dplyr")
library("grid")
library("psych")
library("seqinr")
library("ape")
library("phylobase")
library("RColorBrewer")
library("phytools")
library("geiger")
library("picante")
library("lme4")
library("lmerTest")
library("adephylo")
library("psych")
library("tidyr")
source("~/GitHub/Rpf/bin/Phylo4D.JTL.R")

# SEM
sem <- function(x) {
  sd(x) / sqrt(length(x))
}

# Confidence Hulls
add.hull <- function(model = "", pred.frame = ""){
  CI.U <- predict(model, interval = "c", newdata=pred.frame)[, "upr"]
  CI.L <- predict(model, interval = "c", newdata=pred.frame)[, "lwr"]
  pred.frame2 <- unlist(pred.frame)
  X.Vec <- c(pred.frame2, tail(pred.frame2, 1), rev(pred.frame2),
               head(pred.frame2, 1))
  Y.Vec <- c(CI.U, tail(CI.L, 1), rev(CI.L), head(CI.U,1))
  polygon(X.Vec, Y.Vec, col = "gray90", border = NA)
}
```

## Load and process growth data

```{r}
# Load parameters from Gompertz analysis of growth curves

gc.param.1.dat <- read.delim("./code/5.Generality/data/parms.out.plate1.txt", sep = "\t", header = TRUE)
gc.param.2.dat <- read.delim("code/5.Generality/data/parms.out.plate2.txt", sep = "\t", header = TRUE)

# Load key associated with growth curve experiment 
gc.key <- read.delim("code/5.Generality/data/generality.gc.well.key.txt", sep = "\t", header = TRUE)

# Merge data and key for data sets
key.1 <- subset(gc.key, Run_Num == "1", select = c(Strain, Gram, Phylum, Genus, Treatment, Rep))

gc.param.1 <- cbind(gc.param.1.dat, key.1)

key.2 <- subset(gc.key, Run_Num == "2", select = c(Strain, Gram, Phylum, Genus, Treatment, Rep))

gc.param.2 <- cbind(gc.param.2.dat, key.2)

# Merge data sets
gc.total <- rbind(gc.param.1, gc.param.2)

# Select desired columns
gc <- gc.total[,c(4:6,14:19)]
```

## Perform two-way test for effect of Rpf among strains

```{r}
# Convert Treatment and Strain from integer or character to factors
gc$Strain <- as.factor(gc$Strain)
gc$Treatment <- as.factor(gc$Treatment)
gc$Rep <- as.factor(gc$Rep)
gc$Gram <- as.factor(gc$Gram)
Rpf <- ifelse(gc$Treatment == "Rpf+", "plus", "minus")
gc <- data.frame(gc, Rpf)
gc$Rpf <- as.factor(gc$Rpf)

# Test using generalized linear mixed model (GLMM)
#rpf.glmm.A <- lmer(A ~ Treatment * Strain + (1|Strain), data = gc)
#rpf.glmm.umax <- lmer(umax ~ Treatment * Strain + (1|Strain), data = gc)
#rpf.glmm.lag <- lmer(L ~ Treatment * Strain + (1|Strain), data = gc)


# Two-way ANOVA on yield
rpf.aov.A <- aov(A ~ Treatment + Strain + Treatment*Strain, data = gc)

#                 Df Sum Sq Mean Sq F value   Pr(>F)    
# Treatment         1  0.156  0.1562  47.678 1.65e-09 ***
# Strain           11  5.303  0.4821 147.196  < 2e-16 ***
# Treatment:Strain 11  0.121  0.0110   3.347 0.000916 ***
# Residuals        72  0.236  0.0033                
          
# Two-way ANOVA on umax
rpf.aov.umax <- aov(umax ~ Treatment + Strain + Treatment*Strain, data = gc)

#                 Df  Sum Sq  Mean Sq F value   Pr(>F)    
# Treatment         1 0.00476 0.004759  24.676 4.42e-06 ***
# Strain           11 0.17752 0.016138  83.681  < 2e-16 ***
# Treatment:Strain 11 0.00863 0.000785   4.069 0.000121 ***
# Residuals        72 0.01389 0.000193                   

# Two-way ANOVA on L
rpf.aov.L <- aov(L ~ Treatment + Strain + Treatment*Strain, data = gc)

#                 Df Sum Sq Mean Sq F value Pr(>F)    
# Treatment         1      2     1.7   0.571  0.452    
# Strain           11   7754   704.9 232.443 <2e-16 ***
# Treatment:Strain 11     32     2.9   0.970  0.481    
# Residuals        72    218     3.0                             
```

## Test for Gram + vs. Gram - Rpf effect with generalized linear mixed model (GlMM)

```{r}
# https://www.flutterbys.com.au/stats/tut/tut9.2a.html
# random intercept for Strains (i.e., 1|Strain)
# REML (residual maximum likelihood): helps account for unbalanced design

# Yield
gram.A <- lmer(A ~ Treatment + Gram + Treatment*Gram + 
          (1|Strain), REML = TRUE, data = gc)
gram.A <- update(gram.A) # calls lmerTest package
anova(gram.A) # calculates p-values using Satterthwaite method for denominator df 

# Type III Analysis of Variance Table with Satterthwaite's method
#                   Sum Sq    Mean Sq NumDF DenDF F value Pr(>F)
# Treatment      0.177789 0.177789     1    82 45.6297 1.908e-09 ***
# Gram           0.014658 0.014658     1    10  3.7619   0.08114 .  
# Treatment:Gram 0.036896 0.036896     1    82  9.4694   0.00284 ** 

# Significant Rpf x Gram interaction: (F[1,82]= 9.4694, P = 0.00284)

# Maximum growth rate
gram.umax <- lmer(umax ~ Treatment + Gram + Treatment*Gram + (1|Strain), REML = TRUE, data = gc)
gram.umax <- update(gram.umax)
anova(gram.umax)

#Type III Analysis of Variance Table with Satterthwaite's method
#                 Sum Sq  Mean Sq NumDF DenDF F value    Pr(>F)    
#                  Sum Sq   Mean Sq NumDF DenDF F value    Pr(>F)    
# Treatment      0.0056501 0.0056501     1    82  22.407 9.102e-06 ***
# Gram           0.0022331 0.0022331     1    10   8.856  0.013905 *  
# Treatment:Gram 0.0018395 0.0018395     1    82   7.295  0.008399 ** 

# Significant Rpf x Gram interaction (F[1,82]= 7.295, P = 0.008399)

# Lag time
gram.L <- lmer(L ~ Treatment + Gram + Treatment*Gram + (1|Strain), 
               REML = TRUE, data = gc)
gram.L <- update(gram.L)
anova(gram.L)

# Type III Analysis of Variance Table with Satterthwaite's method
#                  Sum Sq   Mean Sq NumDF DenDF F value    Pr(>F)    
# Treatment      1.4601  1.4601     1    82  0.4781 0.4912
# Gram           5.1503  5.1503     1    10  1.6863 0.2232
# Treatment:Gram 0.2860  0.2860     1    82  0.0936 0.7604

# No effects of Gram or treatment on lag time (P > 0.05)
```

## Calculate effect sizes for growth parameters 

Using Cohen's D, which quantifies effect between two mean values.
Calculated as difference of means divided by pooled standard deviation
Cohen's D of 1.0, indicates groups differ by one standard deviation, etc.
As a rule of thumb, D = |0.2| = "small", |0.5| = "medium", >|0.8| = "large" effects

```{r}
# Calculate means across strains for +Rpf Treatment 
gc.rpf.plus.strains <- filter(gc, Treatment == "Rpf+") %>%
   dplyr::select(A, umax, L, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.A = mean(A), mean.umax = mean(umax), mean.L = mean(L)) 

# Calculate means across strains for -Rpf Treatment 
gc.rpf.minus.strains <- filter(gc, Treatment == "Rpf-") %>%
   dplyr::select(A, umax, L, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.A = mean(A), mean.umax =mean(umax), mean.L = mean(L))

# Calculate pooled standard deviation across strains 
gc.rpf.pool.sd <- dplyr::select(gc, A, umax, L, Strain) %>%
   group_by(Strain) %>%
   summarize(sd.A = sd(A), sd.umax = sd(umax), sd.L = sd(L)) 

# Calculate effect size across all strains with Cohen's D
cohen.d <- (gc.rpf.plus.strains[,2:4] - gc.rpf.minus.strains[,2:4]) / gc.rpf.pool.sd[,2:4]
colnames(cohen.d) <- c("A", "umax", "L")
rownames(cohen.d) <- t(as.data.frame(gc.rpf.minus.strains[1]))

# Summarize Cohen's D across strains
summary.cohen.d <- cohen.d %>%
  summarise(
    mean.A = mean(A), sem_A = sem(A),
    mean.umax = mean(umax), sem_umax = sem(umax),
    mean.L = mean(L), sem_L = sem(L)
  )

#       mean.A     sem_A    mean.umax   sem_umax      mean.L     sem_L
#    0.9989581  0.2814522   0.8458633   0.3223238 -0.7178522 0.3085042


# Calculate effect size for Gram + vs. Gram - using Cohen's D

gram.code <- as.factor(c("neg","pos","pos","neg","pos","neg","neg","neg",
            "pos","pos","pos","pos"))
cohen.d.gram <- data.frame(cohen.d, gram.code)

gc.gram <- dplyr::select(cohen.d.gram, A, umax, L, gram.code) %>%
   group_by(gram.code) %>%
   summarize(
     mean.A = mean(A), sem_A = sem(A),
     mean.umax = mean(umax), sem_umax = sem(umax),
     mean.L = mean(L), sem_L = sem(L)
   )

# gram.code   mean.A  sem_A     mean.umax   sem_umax    mean.L   sem_L
# neg        1.67     0.0934     1.65       0.0780    -0.606    0.501
# pos        0.519    0.391      0.268      0.436     -0.798    0.421

# umax: "large" positive for Gram - (1.65); "small" positive for Gram + (0.27)
# lag: "medium" negative for Gram - (-61); "medium" negative for Gram + (-0.80)
# yield: "large" positive for Gram - (1.65); "medium" positive for  Gram + (0.52)
# Summary: Rpf stimulated Gram - bacteria in terms of higher yield and umax, but longer lag
# Summary: Rpf stimulated Gram + bacteria; same direction as Gram - but weaker
```

## Phylogenetic analyses

# Make alignment

# Using muscle (muscle3.8.31_i86darwin64.tar.gz), I aligned 16S sequences using following command:

```{bash, results = 'hide', eval = FALSE}
#muscle -in ./data/phylogeny/Rpf.16S.tree.fasta-out ./data/phylogeny/Rpf.16S.tree.af
```

# View alignment

```{r}
# Read Alignment File {seqinr}
read.aln <- read.alignment(file = "./data/phylogeny/Rpf.16S.tree.afa", format = "fasta")  

# Convert Alignment File to DNAbin Object {ape}
rpf.DNAbin <- as.DNAbin(read.aln) 

# Identify Base Pair Region of 16S rRNA Gene to Visuzlize
window <- rpf.DNAbin[, 100:500] 

# Command to Visusalize Sequence Alignment {ape}
image.DNAbin(window, cex.lab = 0.50) 
```

# View maximum likelihood tree

```{r}
# Read Alignment File {seqinr}

ml.tree.full <- read.tree("./data/phylogeny/RAxML_bipartitions.rpf.ml")

# Rotate internal node so Gram-positive strains are together
ml.tree <- ape::rotate(ml.tree.full, node = 15)

par(mar = c(1,1,2,1) + 0.1)
plot.phylo(ml.tree, type = "phylogram", direction = "right", show.tip.label=TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1, 
           main = "Maximum Likelihood with Support Values")
add.scale.bar(cex = 0.7)
nodelabels(ml.tree$node.label, font = 2, bg = "white", frame = "r", cex = 0.5)
```

# Map effect sizes onto tree

```{r}
# Keep rooted but drop outgroup branch
tree <- drop.tip(ml.tree.full, "Methanosarcina", )
tree$edge.length <- tree$edge.length + 10^-7

# Reorder growth parameters to match tips
gc.map <- as.matrix(cohen.d[match(tree$tip.label, row.names(cohen.d)), ])

# Subset tree
tree <- keep.tip(tree, row.names(cohen.d))
tree$node.label <- c(1:tree$Nnode)
tree$node.label <- c(13:23)

plot.phylo(tree, type = "phylogram", direction = "right", show.tip.label=TRUE, show.node.label = TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1)

# Rotate node (node number changes after removing outgroup)
tree.2 <- ape::rotate(tree, node = 13)
tree.3 <- reroot(tree.2, node.number = 14)

plot.phylo(tree.3, type = "phylogram", direction = "rightwards",
           show.tip.label=TRUE, show.node.label = TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1)

# Define color palette
BuRd <- c("#0571B0","#92C5DE","gray92", "#F7F7F7","#F4A582","#CA0020")
mypalette <- colorRampPalette(BuRd)

# Make figure
png(filename="~/GitHub/Rpf/figures/Fig5.png",
    width = 1200, height = 1200, res = 96*2, bg = "white")

#par(mar=c(0.5,0.5,0.5,1.5) + 0.2)
par(mar=c(3, 3, 0.5, 1.5) + 0.1, oma = c(0.5, 1, 0, 0))

dat.2 <- cohen.d[c(tree.3$tip.label), ]

x <- phylo4d(tree.3, tip.data = dat.2)
x@edge[2:23, ] <- tree.3$edge
table.phylo4d(x, treetype = "phylogram", 
                symbol = "colors", show.node = F, cex.label = 0.75, 
                grid = FALSE, scale = TRUE, use.edge.length = FALSE, 
                edge.color = "black", edge.width = 2, box = FALSE, 
                col = mypalette(25), pch = 15, cex.symbol = 2, font = 2,
                ratio.tree = 0.75, cex.legend = 1.25, center = FALSE,
                show.var.label = FALSE)

text(3.6, 12.3, expression("Bacillota"), cex = 1.15)
text(3.4, 9.0, expression("Actinomycetota"), cex = 1.15)
#text(0, 4.75, expression(bold("Gram -")), cex = 1)
mtext(expression(bold("Gram -")), side = 2, line = -1.5, at = 4.5, cex = 1.25, las = 2)
mtext(expression(bold("Gram +")), side = 2, line = 0.0, at = 9, cex = 1.25, las = 2)
#text(8.1, 10.7, expression("Corynebacterineae"), cex = 0.65)
#text(7.0, 7.10, expression("Micrococcineae"), cex = 0.65)
text(5.0, 4.8, expression("Bacteroidota"), cex = 1.15)
text(4.8, 2, expression("Proteobacteria"), cex = 1.15)
text(9.5, 3.85, expression(beta), cex= 1)
text(9.5, 2.30, expression(alpha), cex= 1)
text(9.5, 1.30, expression(gamma), cex= 1)

mtext(expression("Lag"), side = 1, line = -1.25, at = 13.25, cex = 1.15, las = 2)
mtext(expression(paste(mu, "max")), side = 1, line = -1.25, at = 12.4, cex = 1.15, las = 2)
mtext(expression("Yield"), side = 1, line = -1.25, at = 11.25, cex = 1.15, las = 2)
mtext(expression("Rpf effect size"), side = 2, line = -7.2, at = -1.1, cex = 1.15, las = 2)
mtext(expression("-2"), side = 2, line = -1.8, at = -0.2, cex = 1.15, las = 2)
mtext(expression("-1"), side = 2, line = -3.75, at = -0.2, cex = 1.15, las = 2)
mtext(expression("1"), side = 2, line = -5.6, at = -0.2, cex = 1.15, las = 2)
mtext(expression("2"), side = 2, line = -7.4, at = -0.2, cex = 1.15, las = 2)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("./figures/Fig5.png")

grid.raster(img)
```

## Phylogenetic tests on effect sizes

```{r}

# Blomberg's K:
# K is a scaled ratio of the variance among species over the contrasts variance (the latter of which will be low if phylogenetic signal is high)
# K, a variance ratio, is rescaled by dividing by the Brownian motion expectation. This gives it the property of having an expected value of 1.0 under Brownian evolution
# Thus, K<1 indicates that closely related species resemble each other less than expected under the Brownian motion model of trait evolution. K>1 means that closely related species are more similar than predicted by the model. In other words, greater values of K suggest strongest effects of phylogenetic signal.

# Load tree
ml.tree <- read.tree("./data/phylogeny/RAxML_bipartitions.rpf.ml")

# Root tree
tree.rooted <- ape::root.phylo(ml.tree, outgroup = "Methanosarcina")

# Drop unnecessary tips
tips_to_drop <- setdiff(tree.rooted$tip.label, row.names(cohen.d))
tree <- ape::drop.tip(tree.rooted, tips_to_drop)

# Reorder the data to match tree tip labels
cohen.d3 <- as.matrix(cohen.d[match(tree$tip.label, row.names(cohen.d)), ])
print(tree$tip.label)
print(row.names(cohen.d3))

# Run test for Blomberg's K for Yield (A)
result_K_A <- phylosig(tree, cohen.d3[, 1], method = "K", test = TRUE) # phylotools
print(result_K_A)
#Phylogenetic signal K :  0.36
# P-value (based on 1000 randomizations) : 0.326
# K = 0.36 consistent with overdispersion of Rpf effect
# But P = 0.326, so not significant

# Run test for Blomberg's K for umax
result_K_umax <- phylosig(tree, cohen.d3[, 2], method = "K", test = TRUE) 
print(result_K_umax)
#Phylogenetic signal K : 0.987
#P-value (based on 1000 randomizations) : 0.018
# K = 0.987  consistent with Brownian motion
# P = 0.018 significant

# Run test for Blomberg's K for lag
result_K_lag <- phylosig(tree, cohen.d3[, 3], method = "K", test = TRUE) 
print(result_K_lag)
# Phylogenetic signal K : 0.20 consistent with overdispersion
# P-value (based on 1000 randomizations) : 0.686 
# But P = 0.686, not significant

# Pagel's Lambda:

# when lambda = 0, there's no signal (polytomy)
# when lambda = 1, consistent with Brownian motion
# 0< lambda<1: intermediate level of phylogenetic signal, not as strong as Brownian
# lamda >1: stronger than Brownian

## Yield (A)
phylosig(tree, cohen.d3[, 1], method = "lambda", test = T)
# lambda = 7.33137e-05, P value = 1.0 ==> no phylogenetic signal

# Maximum growth rate (umax)
phylosig(tree, cohen.d3[, 2], method = "lambda", test = T)
# lambda = 0.999927, P value = 0.0130492 ==> Brownian motion

# Lag time (L)
phylosig(tree, cohen.d3[, 3], method = "lambda", test = T)
# lambda =  7.33137e-05, P value = 1 ==> no phylogenetic signal


# Does Rpf effect size decrease with evolutionary distance to KBS0714?

# Create sequence-based distance matrix {ape}
seq.dist.raw <- dist.dna(rpf.DNAbin , model = "raw", pairwise.deletion = FALSE)
seq.dist.mat <- as.matrix(dist(seq.dist.raw))

# Retrieve the KBS0714 distances, drop outgroup, convert to similarity
seq.dist.KBS0714 <- seq.dist.mat[,7]
seq.dist.KBS0714 <- seq.dist.KBS0714[2:13]
seq.sim.KBS0714 <- 1- seq.dist.KBS0714

# Make data frame and order
seq.sim.KBS0714 <- data.frame(names(seq.sim.KBS0714), seq.sim.KBS0714)
colnames(seq.sim.KBS0714) <- c("Strain", "Similarity")
seq.sim.KBS0714 <- seq.sim.KBS0714[order(seq.sim.KBS0714$Strain),]

# Merge similarity with effect sizes
cohen.d <-as.data.frame(cohen.d)
cohen.d <- cohen.d[order(row.names(cohen.d)), ]
seq.sim.final <- data.frame(cohen.d, seq.sim.KBS0714)

seq.sim.A <- lm(seq.sim.final$A ~ seq.sim.final$Similarity, data = seq.sim.final)
# summary(seq.sim.A)
# r2 = 0.01, P = 0.3223 

seq.sim.umax <- lm(seq.sim.final$umax ~ seq.sim.final$Similarity, 
        data = seq.sim.final)
# summary(seq.sim.umax)
# r2 = 0.08, P = 0.20

seq.sim.L <- lm(seq.sim.final$L ~ seq.sim.final$Similarity, data = seq.sim.final)
# summary(seq.sim.L)
# plot(seq.sim.final$umax,seq.sim.final$Similarity)
# r2 = 0.15, P = 0.11

# summary: no evidence that Rpf effect is related to genetic distance to KBS0714
```


# Is the Rpf effect size related to functional traits? (Lennon et al. 2012)

```{r}
# Load trait data
g.traits <- read.delim("data/generality/traits.txt", sep = "\t", header = TRUE)
g.traits <- cbind(g.traits, cohen.d[,1:3])
colnames(g.traits)[14:16] <- c("A.effect", "umax.effect", "lag.effect")
g.traits <- g.traits[!(g.traits$strain =="KBS0712"),]
cor <- corr.test(g.traits[,2:16], method = "pearson")

# Yield effect: 
#  Wopt:    p = 0.02766552, r =  0.658248346
#  Mpamin:  p = 0.01441242, r =  0.70978535
#  breadth: p = 0.07872851, r = -0.551347047 

# umax effect: 
# Wopt:     p = 0.01477756, r =  0.707979447  
# breadth:  p = 0.01258031, r = -0.719381527
# Mpamin:   p = 0.01996483, r =  0.68523424

# Lag effect
# biofilm: p = 0.03578486,  r =  0.6350543
fit.umax.biofilm <- lm(lag.effect ~ biofilm, data = g.traits)
# F[1,9] = 6.083, p-value: 0.03578, r2 = 0.40
# plot(g.traits$biofilm, g.traits$lag.effect)
# Rpf lag effect increases with biofilm
# Strains that don't make biofilms respond faster to Rpf
# Could reflect energetic trade-offs
# Or, if you're in a biofilm, maybe you're less connected to enzymes in environment

# Make multi-panel plot

# Set options
options(timeout = 600) # prevent timing out

png(filename="~/GitHub/Rpf/figures/Fig6.png",
    width = 1200, height = 1600, res = 96*2)
plot.new()
par(oma=c(6, 6, 5, 5), mar=c(2, 4, 2, 4), mfrow=c(3, 2))
par(mai=c(0.7, 0.4, 0.1, 0))

##########################
# Panel 1: MPamin for umax 
##########################

# calculate regressions: umax effect vs. Mpamin
fit.umax.mpamin <- lm(umax.effect ~ Mpamin, data = g.traits)
pred.frame.umax.mpamin <- data.frame(Mpamin = seq(-2.4, -0.01, by = 0.05))
# F(1,9) = 3.023, p-value: 0.01441, r2, = 0.50

# create minimum plot 
plot(Mpamin ~ umax.effect, g.traits, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(-2.6, 0), ylim = c(-3.5, 3.5), las = 1)
box(lwd = 2)

mtext(expression(bold(mu[max])), side = 2, line = -10, at = 6, cex = 2, las = 2)

# Add hull confidence hulls and regression line
add.hull(fit.umax.mpamin, pred.frame.umax.mpamin)
matlines(pred.frame.umax.mpamin, predict(fit.umax.mpamin, 
      interval = "c", newdata = pred.frame.umax.mpamin), lty = c(3, 3, 3), 
      lwd = c(2, 0, 0), col = "red")

# Add points
points(umax.effect ~ Mpamin, g.traits,
       pch = 21, col = "gray42", bg = "gray42", lwd = 2, cex = 1.5)

# Add labels
mtext("Minimum moisture (MPa)", side = 1, line = 3.0, cex = 1.25, at = 0.25)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-3", "0", "3"), at = c(-3, 0, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-3, 0, 3), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-2.5", "-2", "-1.5", "-1", "-0.5", "0"), 
     at = c(-2.5,-2,-1.5,-1,-0.5,0))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(-2.5,-2,-1.5,-1,-0.5,0), labels = F)

# Add stats and panel labels
#text(-0.1, 3, labels = "B", col = "black", cex = 2)
text(-2.2, 3, labels = expression(r^2 == 0.47), cex = 1.25)
text(-2.1, 2, labels = expression(italic("P")~" = 0.020"), cex = 1.25)

##########################
# Panel 2: MPamin for yield 
##########################

# calculate regressions: yield effect vs. Mpamin
fit.A.mpamin <- lm(A.effect ~ Mpamin, data = g.traits)
pred.frame.A.mpamin <- data.frame(Mpamin = seq(-2.4, -0.01, by = 0.05))
# F(1,9) = 9.138, p-value: 0.01441, r2, = 0.50

# create minimum plot 
plot(Mpamin ~ A.effect, g.traits, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(-2.6, 0), ylim = c(-3.5, 3.5), las = 1)
box(lwd = 2)

mtext(expression(bold(Yield)), side = 2, line = -10, at = 6, cex = 1.75, las = 2)

# Add hull confidence hulls and regression line
add.hull(fit.A.mpamin, pred.frame.A.mpamin)
matlines(pred.frame.A.mpamin, predict(fit.A.mpamin, 
      interval = "c", newdata = pred.frame.A.mpamin), lty = c(3, 3, 3), 
      lwd = c(2, 0, 0), col = "red")

# Add points
points(A.effect ~ Mpamin, g.traits,
       pch = 21, col = "gray42", bg = "gray42", lwd = 2, cex = 1.5)

# Add labels

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-3, 0, 3), labels = F)
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-3, 0, 3), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-2.5", "-2", "-1.5", "-1", "-0.5", "0"), 
     at = c(-2.5,-2,-1.5,-1,-0.5,0))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(-2.5,-2,-1.5,-1,-0.5,0), labels = F)

r2<- sprintf("%.2f", 0.50)
text(-2.2, 3, labels = bquote(r^2 == .(r2)), cex = 1.25)
text(-2.1, 2, labels = expression(italic("P")~" = 0.014"), cex = 1.25)

########################################
# Panel 3: niche breadth for umax effect
########################################

# calculate regressions
fit.umax.b <- lm(umax.effect ~ b, data = g.traits)
pred.frame.umax.b <- data.frame(b = seq(-0.16, 1.28, by = 0.1))
# F(1,9) =  9.653, p-value: 0.01258, r2, = 0.52 

# create minimum plot 
plot(b ~ umax.effect, g.traits, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(-0.2, 1.25), ylim = c(-3.5, 3.5), las = 1)
box(lwd = 2)

# Add hull confidence hulls and regression line
add.hull(fit.umax.b, pred.frame.umax.b)
matlines(pred.frame.umax.b, predict(fit.umax.b, 
      interval = "c", newdata = pred.frame.umax.b), lty = c(3, 3, 3), 
      lwd = c(2, 0, 0), col = "red")

# Add points
points(umax.effect ~ b, g.traits,
       pch = 21, col = "gray42", bg = "gray42", lwd = 2, cex = 1.5)

# Add labels
mtext("Niche breadth (MPa)", side = 1, line = 3.0, cex = 1.25, at = 1.5)

mtext(expression('Rpf effect size'), side = 2,
      outer = FALSE, cex = 2.5, line = 5, adj = 0.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-3", "0", "3"), at = c(-3, 0, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-3, 0, 3), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "0.5", "1"), at = c(0, 0.5, 1))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0, 0.5, 1), labels = F)

# Add stats and panel labels
text(0.05, -2, labels = expression(r^2 == 0.46), cex = 1.25)
text(0.09, -2.85, labels = expression(italic("P")~" = 0.013"), cex = 1.25)

#######################################
# Panel 4: niche breadth for lag effect
#######################################

# calculate regressions
fit.A.b <- lm(A.effect ~ b, data = g.traits)
pred.frame.A.b <- data.frame(b = seq(-0.16, 1.28, by = 0.1))
# F(1,9) = 3.931, p-value: 0.07873, r2, = 0.30 

# create minimum plot 
plot(b ~ A.effect, g.traits, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(-0.2, 1.25), ylim = c(-3.5, 3.5), las = 1)
box(lwd = 2)

# Add hull confidence hulls and regression line
add.hull(fit.A.b, pred.frame.A.b)
matlines(pred.frame.A.b, predict(fit.A.b, 
      interval = "c", newdata = pred.frame.A.b), lty = c(3, 3, 3), 
      lwd = c(2, 0, 0), col = "red")

# Add points
points(A.effect ~ b, g.traits,
       pch = 21, col = "gray42", bg = "gray42", lwd = 2, cex = 1.5)

# Major Axes
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-3, 0, 3), labels = F)
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-3, 0, 3), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "0.5", "1"), at = c(0, 0.5, 1))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0, 0.5, 1), labels = F)

# Add stats and panel labels
text(0.05, -2, labels = expression(r^2 == 0.30), cex = 1.25)
text(0.09, -2.85, labels = expression(italic("P")~" = 0.079"), cex = 1.25)

################################
# Panel 5: Wopt for umax effect
################################

# calculate regressions
fit.umax.Wopt <- lm(umax.effect ~ Wopt, data = g.traits)
pred.frame.umax.Wopt <- data.frame(Wopt = seq(-1.25, 0.2, by = 0.1))
# F(1,9) = 9.05, p-value: 0.01478, r2, = 0.50

# create minimum plot 
plot(Wopt ~ umax.effect, g.traits, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.2, -1.25), ylim = c(-3.5, 3.5), las = 1)
box(lwd = 2)

# Add hull confidence hulls and regression line
add.hull(fit.umax.Wopt, pred.frame.umax.Wopt)
matlines(pred.frame.umax.Wopt, predict(fit.umax.Wopt, 
      interval = "c", newdata = pred.frame.umax.Wopt), lty = c(3, 3, 3), 
      lwd = c(2, 0, 0), col = "red")

# Add points
points(umax.effect ~ Wopt, g.traits,
       pch = 21, col = "gray42", bg = "gray42", lwd = 2, cex = 1.5)

mtext("Niche optimum (MPa)", side = 1, line = 3.0, cex = 1.25, at = -1.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("-3", "0", "3"), at = c(-3, 0, 3))
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-2, 0, 2), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "-0.5", "-1"), at = c(0, -0.5, -1))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0, -0.5, -1), labels = F)

# Add stats and panel labels
#text(-1.2, 3, labels = "E", col = "black", cex = 2)
r2<- sprintf("%.2f", 0.50)
text(-0.02, -2, labels = bquote(r^2 == .(r2)), cex = 1.25)
text(-0.06, -2.85, labels = expression(italic("P")~" = 0.015"), cex = 1.25)

###############################
# Panel 6: Wopt for lag effect
###############################

# calculate regressions
fit.A.Wopt <- lm(A.effect ~ Wopt, data = g.traits)
pred.frame.A.Wopt <- data.frame(Wopt = seq(-1.25, 0.2, by = 0.1))
# F(1,9) = 6.881, p-value: 0.02767, r2, = 0.43

# create minimum plot 
plot(Wopt ~ A.effect, g.traits, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.2, -1.25), ylim = c(-3.5, 3.5), las = 1)
box(lwd = 2)

# Add hull confidence hulls and regression line
add.hull(fit.A.Wopt, pred.frame.A.Wopt)
matlines(pred.frame.A.Wopt, predict(fit.A.Wopt, 
      interval = "c", newdata = pred.frame.A.Wopt), lty = c(3, 3, 3), 
      lwd = c(2, 0, 0), col = "red")

# Add points
points(A.effect ~ Wopt, g.traits,
       pch = 21, col = "gray42", bg = "gray42", lwd = 2, cex = 1.5)

# Major Axes
axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-2, 0, 2), labels = F)
axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at=c(-2, 0, 2), labels = F)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "-0.5", "-1"), at = c(0, -0.5, -1))
axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0, -0.5, -1), labels = F)

# Add stats and panel labels
#text(-1.2, 3, labels = "F", col = "black", cex = 2)
text(-0.02, -2, labels = expression(r^2 == 0.43), cex = 1.25)
text(-0.06, -2.85, labels = expression(italic("P")~" = 0.028"), cex = 1.25)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("~/GitHub/Rpf/figures/Fig6.png")
grid.raster(img)
```


## Supplementary figure corresponding to two-way ANOVA
Plot mean and SEM for growth parameter (yield, umax, lag) on each strain
in both +Rpf and -Rpf treatments. 

# Subset data

```{r}
# Order df.table

sort.strain <- c("KBS0812", "KBS0724", "KBS0706", "KBS0702", "KBS0703", 
                 "KBS0714", "KBS0715", "KBS0701", "KBS0711", "KBS0712", 
                 "KBS0710", "KBS0705")

gc.2 <- gc %>%
  mutate(Strain =  factor(Strain, levels = sort.strain)) %>%
  arrange(Strain) 

# Yield (A) for +Rpf 
A.plus <- filter(gc.2, Treatment == "Rpf+") %>%
   dplyr::select(A, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.A = mean(A), SE.A = sem(A))

# Yield (A) for -Rpf 
A.minus <- filter(gc.2, Treatment == "Rpf-") %>%
   dplyr::select(A, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.A = mean(A), SE.A = sem(A))

# Max growth (umax) for +Rpf 
umax.plus <- filter(gc.2, Treatment == "Rpf+") %>%
   dplyr::select(umax, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.umax = mean(umax), SE.umax = sem(umax)) 

# Max growth (umax) for -Rpf 
umax.minus <- filter(gc.2, Treatment == "Rpf-") %>%
   dplyr::select(umax, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.umax = mean(umax), SE.umax = sem(umax)) 

# Lag time (L) for +Rpf 
L.plus <- filter(gc.2, Treatment == "Rpf+") %>%
   dplyr::select(L, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.L = mean(L), SE.L = sem(L)) 

# Lag time (L) for -Rpf 
L.minus <- filter(gc.2, Treatment == "Rpf-") %>%
   dplyr::select(L, Strain) %>%
   group_by(Strain) %>%
   summarize(mean.L = mean(L), SE.L = sem(L)) 
```

# Make figure

```{r}
png(filename="~/GitHub/Rpf/figures/FigS6.png",
    width = 900, height = 1200, res = 96*2, bg = "white")

plot.new()
par(oma=c(9, 5, 0, 0), mar=c(1, 2, 1, 1), mfrow = c(3, 1))
 
# Created x-axis plotting locations
x.Rpf.minus <- seq(from = 1, to = 46, by = 4)
x.Rpf.plus <- seq(from = 2, to = 46, by = 4)

# Plot means for yield
plot(x.Rpf.minus, A.minus$mean.A, ylim = c(0, 2.5), xlim = c(0, 46), 
     pch = 22, col = "black", bg = "white", lwd = 1, cex = 2.25, yaxt = "n", 
     xaxt = "n", cex.lab = 2, cex.axis = 1.5, las = 1, ylab = "", xlab = "")
      box(lwd = 2)
      
points(x.Rpf.plus, A.plus$mean.A, pch = 22, col = "black", bg = "grey", 
       lwd = 1, cex = 2.25)

# Plot -Rpf errors
arrows(x0 = x.Rpf.minus, y0 = A.minus$mean.A, 
       y1 = (A.minus$mean.A + A.minus$SE.A), angle = 90, length = 0.02, lwd = 1.5)

arrows(x0 = x.Rpf.minus, y0 = A.minus$mean.A, 
       y1 = (A.minus$mean.A - A.minus$SE.A), angle = 90, length = 0.02, lwd = 1.5)

# Plot +Rpf errors
arrows(x0 = x.Rpf.plus, y0 = A.plus$mean.A, 
       y1 = (A.plus$mean.A + A.plus$SE.A), angle = 90, length = 0.02, lwd = 1.5)

arrows(x0 = x.Rpf.plus, y0 = A.plus$mean.A, 
       y1 = (A.plus$mean.A - A.plus$SE.A), angle = 90, length = 0.02, lwd = 1.5)

# Major Axes

x.int <-seq(from = 1.5, to = 46, by = 4)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = F, at = x.int, las =3)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.0", "0.5", "1.0", "1.5", "2.0", "2.5"), 
     at = c(0, 0.5, 1.0, 1.5, 2.0, 2.5))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0, 0.5, 1.0, 1.5, 2.0, 2.5), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = x.int, labels = F)

mtext(expression('Yield (OD600)'), side = 2,
      outer = FALSE, cex = 1.25, line = 4, adj = 0.5)

# Add p-values
text(36, 2.4, labels = expression("Treatment x Strain:" ~italic("P")~"= 0.0009"), cex = 1)

legend(1, 2.6, c("-Rpf", "+Rpf"), pch = 22, pt.bg = c("white", "grey"), 
       pt.cex = 2, pt.lwd = 1, bty = 'n', y.intersp = 1)


# Plot means for umax
plot(x.Rpf.minus, umax.minus$mean.umax, ylim = c(0.01, 0.21), xlim = c(0, 46), 
     pch = 22, col = "black", bg = "white", lwd = 1, cex = 2.25, yaxt = "n", 
     xaxt = "n", cex.lab = 2, cex.axis = 1.5, las = 1, ylab = "", xlab = "")
      box(lwd = 2)
      
points(x.Rpf.plus, umax.plus$mean.umax, pch = 22, col = "black", bg = "grey", 
       lwd = 1, cex = 2.25)

# Plot -Rpf errors
arrows(x0 = x.Rpf.minus, y0 = umax.minus$mean.umax, 
       y1 = (umax.minus$mean.umax + umax.minus$SE.umax), angle = 90, length = 0.02, lwd = 1.5)

arrows(x0 = x.Rpf.minus, y0 = umax.minus$mean.umax, 
       y1 = (umax.minus$mean.umax - umax.minus$SE.umax), angle = 90, length = 0.02, lwd = 1.5)

# Plot +Rpf errors
arrows(x0 = x.Rpf.plus, y0 = umax.plus$mean.umax, 
       y1 = (umax.plus$mean.umax + umax.plus$SE.umax), angle = 90, length = 0.02, lwd = 1.5)

arrows(x0 = x.Rpf.plus, y0 = umax.plus$mean.umax, 
       y1 = (umax.plus$mean.umax - umax.plus$SE.umax), angle = 90, length = 0.02, lwd = 1.5)

# Major Axes
x.int <-seq(from = 1.5, to = 46, by = 4)
axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = F, at = x.int, las =3)

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0.02", "0.08", "0.14", "0.2"), 
     at = c(0.02, 0.08, 0.14, 0.2))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0.02, 0.08, 0.14, 0.2), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = x.int, labels = F)
  
mtext(expression(mu [max]* ' (hr'^-1*')'), side = 2,
 outer = FALSE, cex = 1.25, line = 4, adj = 0.5)

# Add p-values
text(36, 0.02, labels = expression("Treatment x Strain:" ~italic("P")~"= 0.0001"), 
     cex = 1)


# Plot means for Lag
plot(x.Rpf.minus, L.minus$mean.L, ylim = c(0, 45), xlim = c(0, 46), 
     pch = 22, col = "black", bg = "white", lwd = 1, cex = 2.25, yaxt = "n", 
     xaxt = "n", cex.lab = 2, cex.axis = 1.5, las = 1, ylab = "", xlab = "")
      box(lwd = 2)
      
points(x.Rpf.plus, L.plus$mean.L, pch = 22, col = "black", bg = "grey", 
       lwd = 1, cex = 2.25)

# Plot -Rpf errors
arrows(x0 = x.Rpf.minus, y0 = L.minus$mean.L, 
       y1 = (L.minus$mean.L + L.minus$SE.L), angle = 90, length = 0.02, lwd = 1.5)

arrows(x0 = x.Rpf.minus, y0 = L.minus$mean.L, 
       y1 = (L.minus$mean.L - L.minus$SE.L), angle = 90, length = 0.02, lwd = 1.5)

# Plot +Rpf errors
arrows(x0 = x.Rpf.plus, y0 = L.plus$mean.L, 
       y1 = (L.plus$mean.L + L.plus$SE.L), angle = 90, length = 0.02, lwd = 1.5)

arrows(x0 = x.Rpf.plus, y0 = L.plus$mean.L, 
       y1 = (L.plus$mean.L - L.plus$SE.L), angle = 90, length = 0.02, lwd = 1.5)

# Major Axes
x.int <-seq(from = 1.5, to = 46, by = 4)

axis(side = 1, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = sort.strain, at = x.int, las =3, col.axis = "gray40")

axis(side = 2, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     labels = c("0", "10", "20", "30","40"), at = c(0, 10, 20, 30, 40))

axis(side = 4, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = c(0, 10, 20, 30, 40), labels = F)

axis(side = 3, lwd.ticks = 2, cex.axis = 1.25, las = 1,
     at = x.int, labels = F)

mtext(expression('Lag time (hr)'), side = 2,
      outer = FALSE, cex = 1.25, line = 4, adj = 0.5)

# Add p-values
text(40, 42, labels = expression("Strain:" ~italic("P")~"< 0.0001"), cex = 1)
#text(4.2, 33, labels = expression("Treatment:" ~italic("P")~"< 0.0001"), cex = 1)

# Add taxon names and lines
mtext('Gram +', side = 1, line = 8, at = 15, cex = 1.25)
mtext('Gram -', side = 1, line = 8, at = 37, cex = 1.25)

par(xpd=NA)
segments(1, -35, 26, -35, col = "black", lwd = 2)
segments(29, -35, 46, -35, col = "black", lwd = 2)

# Close Plot Device
dev.off()
graphics.off()

# Show Plot
img <- readPNG("./figures/FigS6.png")

grid.raster(img)
```
